[sqlancer] result sets mismatch

<!-- (At least include the following, feel free to add if you have more content) -->

### Steps to reproduce the behavior (Required)
```
DROP DATABASE IF EXISTS w14;
CREATE DATABASE w14;
USE w14;
admin set frontend config ("enable_decimal_v3" = "true");
CREATE TABLE IF NOT EXISTS t0 (c_0_0 DATE NULL ,c_0_1 INT NOT NULL ,c_0_2 DATE NOT NULL ,c_0_3 INT NOT NULL ,c_0_4 BIGINT NOT NULL ,c_0_5 CHAR(1) NOT NULL ,c_0_6 INT NOT NULL ,c_0_7 DOUBLE REPLACE NULL ,c_0_8 DATE REPLACE_IF_NOT_NULL NULL ,c_0_9 HLL HLL_UNION NOT NULL ,c_0_10 VARCHAR(21) MAX NOT NULL ,c_0_11 BOOLEAN REPLACE NOT NULL ,c_0_12 BOOLEAN REPLACE NULL ) AGGREGATE KEY (c_0_0,c_0_1,c_0_2,c_0_3,c_0_4,c_0_5,c_0_6) PARTITION BY RANGE(c_0_0) (START ("2010-01-01") END ("2022-12-31") EVERY (INTERVAL 474 day)) DISTRIBUTED BY HASH (c_0_5,c_0_0) BUCKETS 3 properties("replication_num"="3", "bloom_filter_columns"="c_0_4, c_0_5") ;
CREATE TABLE t1 (c_1_0 DATE NOT NULL,c_1_1 ARRAY<INT> NOT NULL,c_1_2 TINYINT NOT NULL,c_1_3 BIGINT NOT NULL,c_1_4 FLOAT NOT NULL,c_1_5 BOOLEAN NOT NULL,c_1_6 DATE NOT NULL,c_1_7 DATETIME NOT NULL,c_1_8 FLOAT NOT NULL,c_1_9 ARRAY<INT> NOT NULL, INDEX index1_c_1_2(`c_1_2`) USING BITMAP, INDEX index1_c_1_5(`c_1_5`) USING BITMAP) PRIMARY KEY (c_1_0) DISTRIBUTED BY HASH (c_1_0) BUCKETS 3 properties("replication_num"="3") ;
CREATE TABLE IF NOT EXISTS t2 (c_2_0 DATETIME NOT NULL ,c_2_1 DATETIME NOT NULL ,c_2_2 BOOLEAN NOT NULL ,c_2_3 BOOLEAN NULL ,c_2_4 INT NOT NULL ,c_2_5 DECIMAL(23, 14)  NOT NULL ,c_2_6 DATETIME NULL ,c_2_7 DECIMAL(24, 14)  NOT NULL ,c_2_8 DATETIME NOT NULL ,c_2_9 DATETIME NOT NULL ,c_2_10 BOOLEAN NULL ,c_2_11 LARGEINT NOT NULL ,c_2_12 CHAR(1) NULL ,c_2_13 BOOLEAN NOT NULL ,c_2_14 FLOAT NOT NULL ) UNIQUE KEY (c_2_0,c_2_1,c_2_2,c_2_3,c_2_4,c_2_5,c_2_6,c_2_7,c_2_8,c_2_9,c_2_10,c_2_11,c_2_12,c_2_13) PARTITION BY RANGE(c_2_0) (START ("2010-01-01") END ("2022-12-31") EVERY (INTERVAL 135 day)) DISTRIBUTED BY HASH (c_2_0,c_2_5,c_2_9,c_2_12,c_2_8,c_2_7,c_2_10,c_2_6,c_2_3,c_2_4,c_2_13,c_2_1,c_2_11,c_2_2) BUCKETS 3 properties("replication_num"="3") ;
INSERT INTO t0 VALUES ('2010-01-07', 19, '2010-01-27', 22, -1827972752, '', 21, 0.07082692083695852, '2010-01-23', hll_hash('Tv'), '4', true, false);
INSERT INTO t1 VALUES ('2010-01-11', [-1156292347, -1827972752, -1247080232, -1247080232, -266824351, -1827972752, -1156292347, -558033645], 0, 891232616, 0.0, false, '2010-01-18', '2010-01-26 08:00:14', 0.6943859966940347, [891232616, -1815527098, -558033645, -1741136704, -717183377, -1815527098, -1156292347, 223287925, -1629298125, -1221024121, 1844488466, -558033645, 992510042]), ('2010-01-15', [992510042, -1019167431, -558033645, 223287925, -365577527], 6, -1838737366, 0.6564618384496556, true, '2010-01-18', '2010-01-24 08:00:14', 0.25729411329574947, [992510042, 231212788, -365577527, 1854123270, 2017440113, -1344340797, -1657180761, 223287925]);
INSERT INTO t0 VALUES ('2010-01-25', 19, '2010-01-31', 3, 1638194158, '', 26, NULL, '2010-01-10', hll_hash('q72KO'), '1ECPbq', false, false), ('2010-01-22', 20, '2009-12-31', 8, -1247080232, '', 12, 0.20227814987197568, '2010-01-06', hll_hash('Su'), 'Q9227QgUB', true, true);
INSERT INTO t0 VALUES ('2010-01-17', 5, '2010-01-28', 6, 1056630101, '', 21, 1.85412327E9, '2010-01-12', hll_hash('LS'), 'n4', true, false), ('2010-01-06', 13, '2010-01-12', 11, 1200107599, '', 22, 0.8534822574132734, '2010-01-02', hll_hash('eR'), 'HybPdN', true, true);
INSERT INTO t1 VALUES ('2010-01-10', [1894557683, 1659126514, -1019167431, -1579105438, -544356686, -320828620, -459361351, 1894557683, 2121248435, -1579105438, 650019716, 650019716, -1344340797, -1939533430, 1056630101], 1, -1657180761, 0.6564618384496556, true, '2010-01-26', '2010-01-24 08:00:14', 0.11528827854346646, [-1815527098, 1844488466, -1579105438, -676419865, -606106711, 1898743517, 1613351555, 1771378049, 864744394, 223287925, 1132772386, 1919958489, -1502819399, -419662396]), ('2010-01-28', [107357866, -473403990, 2080425696, 507072568, -1815527098, 381922959], 0, -211601319, 0.14186443392800308, false, '2010-01-28', '2010-01-31 08:00:14', 0.3164918717149112, [107357866, 223287925, 1091926990, -606106711, 348398562]), ('2010-01-19', [-459361351, -732696831, -493688971, 107357866, -1657487788, 907388091, 1279292082, -1657180761, -2137932069, 1661623916, -580648924], 0, 982432285, 0.6943859966940347, true, '2010-01-04', '2010-01-14 08:00:14', 0.8910568366441952, [1797068906, 1840820514, -493688971, -1543093410, -367335475, -1504365611, 1126481588, -1320127530, -312737929, 1724178828, 1661623916, -1940486154, 1371474089, 1093385425]);
INSERT INTO t2 VALUES ('2010-01-23 08:00:14', '2010-01-07 08:00:14', true, true, 2, 0.00000765302482, '2010-01-15 08:00:14', 0.00003291490881, '2010-01-12 08:00:14', '2010-01-09 08:00:14', false, 992510042, '', false, 0.9039010069712182), ('2010-01-27 08:00:14', '2010-01-09 08:00:14', false, false, 30, 0.00000115101140, '2010-01-02 08:00:14', 0.00004482765937, '2010-01-27 08:00:14', '2010-01-21 08:00:14', true, -422576195, '', true, 8.64744394E8);
INSERT INTO t2 VALUES ('2010-01-25 08:00:14', '2010-01-09 08:00:14', false, false, 15, 0.00000638796052, '2010-01-08 08:00:14', 0.00002921352457, '2010-01-07 08:00:14', '2010-01-15 08:00:14', true, 1259134481, '', true, 0.9582557010534746), ('2010-01-15 08:00:14', '2010-01-08 08:00:14', false, true, 31, 0.00000797459250, '2010-01-19 08:00:14', 0.00005101225428, '2010-01-09 08:00:14', '2009-12-31 08:00:14', false, -1333039082, '', true, 0.20612683181841418);
INSERT INTO t2 VALUES ('2010-01-02 08:00:14', '2010-01-16 08:00:14', false, false, 9, 1.7578603E-7, '2010-01-22 08:00:14', 0.00009341318055, '2010-01-27 08:00:14', '2010-01-05 08:00:14', true, -1967514451, '', true, 0.14879674236801266), ('2010-01-07 08:00:14', '2010-01-08 08:00:14', true, false, 20, 0.00000961378948, '2010-01-10 08:00:14', 0.00004617058290, '2010-01-21 08:00:14', '2010-01-29 08:00:14', true, 32431337, '', true, 0.41800186995976774);
CREATE VIEW v0 AS SELECT DISTINCT t2_154.c_2_0, t2_154.c_2_1, t2_154.c_2_2, t2_154.c_2_3, t2_154.c_2_4, t2_154.c_2_5, t2_154.c_2_6, t2_154.c_2_7, t2_154.c_2_8, t2_154.c_2_9, t2_154.c_2_10, t2_154.c_2_11, t2_154.c_2_12, t2_154.c_2_13, t2_154.c_2_14 FROM t2 AS t2_154 WHERE (NOT (false)) ORDER BY t2_154.c_2_12, t2_154.c_2_0, t2_154.c_2_10, t2_154.c_2_5;
INSERT INTO t1 VALUES ('2010-01-28', [507072568, 1371474089, -2071237034], 7, -2013548003, 0.41800186995976774, false, '2010-01-15', '2010-01-27 08:00:14', 0.11528827854346646, [-95879602, -1815527098, 1613351555, -587646870, -1838737366, 1926566181, 60415161, -1540507567]), ('2010-01-06', [-638313089, 1910735030, -558033645, -812897659, -1056312871, 2124867089, 1001070037], 0, 1816475271, -1.940486154E9, false, '2010-01-08', '2010-01-06 08:00:14', -6.76419865E8, [231212788, 1272183614, -253927132, -1566067023, -1019167431, -558033645, 71278258, 1990928110, -828793786, -253927132, 768675271, 1200107599, 1272202242, -1625813675]);
INSERT INTO t0 VALUES ('2010-01-05', 1, '2010-01-29', 5, 976666134, '', 8, 0.10032574922578519, '2010-01-18', hll_hash('8t'), 's', false, NULL);
SELECT v0_162.c_2_0 FROM v0 AS v0_162;
---- cardinality: 6
SELECT v0_162.c_2_0 FROM v0 AS v0_162 WHERE ((REVERSE('') ) IS NOT NULL) UNION ALL SELECT v0_162.c_2_0 FROM v0 AS v0_162 WHERE (NOT (((REVERSE('') ) IS NOT NULL))) UNION ALL SELECT v0_162.c_2_0 FROM v0 AS v0_162 WHERE ((((REVERSE('') ) IS NOT NULL)) IS NULL);
---- cardinality: 18

```


```
DROP DATABASE IF EXISTS h9;
CREATE DATABASE h9;
USE h9;
admin set frontend config ("enable_decimal_v3" = "true");
CREATE TABLE IF NOT EXISTS t0 (c_0_0 DATETIME NOT NULL,c_0_1 DATETIME NOT NULL,c_0_2 DATETIME NOT NULL,c_0_3 DATE NOT NULL,c_0_4 LARGEINT NOT NULL,c_0_5 DATETIME NOT NULL,c_0_6 ARRAY NOT NULL,c_0_7 VARCHAR(31) NOT NULL,c_0_8 ARRAY NOT NULL,c_0_9 BOOLEAN NOT NULL,c_0_10 DECIMAL(17, 11) NOT NULL,c_0_11 ARRAY NOT NULL,c_0_12 INT NOT NULL, INDEX index1_c_0_2(c_0_2) USING BITMAP) PRIMARY KEY (c_0_0,c_0_1,c_0_2,c_0_3,c_0_4) PARTITION BY RANGE(c_0_0) (START ("2010-01-01") END ("2022-12-31") EVERY (INTERVAL 1186 day)) DISTRIBUTED BY HASH (c_0_0) BUCKETS 3 properties("replication_num"="3") ;
CREATE TABLE t1 (c_1_0 SMALLINT NOT NULL ,c_1_1 DATETIME NOT NULL ,c_1_2 DECIMAL(8, 5) NOT NULL ,c_1_3 DATETIME NULL ,c_1_4 BOOLEAN NOT NULL ,c_1_5 DATE NULL ,c_1_6 DATE NOT NULL ,c_1_7 SMALLINT NOT NULL ,c_1_8 DATETIME NOT NULL ,c_1_9 DECIMAL(29, 13) NULL ,c_1_10 CHAR(11) NULL ,c_1_11 DATE MAX NOT NULL ,c_1_12 SMALLINT MIN NULL ,c_1_13 BOOLEAN REPLACE NULL ,c_1_14 HLL HLL_UNION NOT NULL ,c_1_15 BOOLEAN REPLACE NOT NULL ,c_1_16 VARCHAR(11) MIN NOT NULL ,c_1_17 DOUBLE REPLACE NULL , INDEX index1_c_1_3(c_1_3) USING BITMAP, INDEX index1_c_1_8(c_1_8) USING BITMAP) AGGREGATE KEY (c_1_0,c_1_1,c_1_2,c_1_3,c_1_4,c_1_5,c_1_6,c_1_7,c_1_8,c_1_9,c_1_10) PARTITION BY RANGE(c_1_5) (START ("2010-01-01") END ("2022-12-31") EVERY (INTERVAL 110 day)) DISTRIBUTED BY HASH (c_1_2,c_1_3,c_1_8,c_1_7) BUCKETS 3 properties("replication_num"="3") ;
CREATE TABLE IF NOT EXISTS t2 (c_2_0 TINYINT NULL ,c_2_1 JSON NULL ,c_2_2 TINYINT NULL ,c_2_3 LARGEINT NOT NULL ,c_2_4 DECIMAL(35, 0) NOT NULL ,c_2_5 ARRAY NULL ,c_2_6 VARCHAR(11) NOT NULL ) DUPLICATE KEY (c_2_0) DISTRIBUTED BY HASH (c_2_0) BUCKETS 3 properties("replication_num"="3") ;
CREATE TABLE t3 (c_3_0 DATETIME NOT NULL ,c_3_1 BOOLEAN REPLACE NOT NULL ,c_3_2 DATETIME REPLACE_IF_NOT_NULL NULL ,c_3_3 PERCENTILE PERCENTILE_UNION NOT NULL ,c_3_4 BOOLEAN REPLACE NOT NULL ,c_3_5 BOOLEAN REPLACE NOT NULL ,c_3_6 DECIMAL(24, 9) SUM NOT NULL ,c_3_7 BIGINT MAX NOT NULL ,c_3_8 PERCENTILE PERCENTILE_UNION NOT NULL ,c_3_9 SMALLINT REPLACE NOT NULL ,c_3_10 FLOAT MIN NOT NULL ) AGGREGATE KEY (c_3_0) PARTITION BY RANGE(c_3_0) (START ("2010-01-01") END ("2022-12-31") EVERY (INTERVAL 96 day)) DISTRIBUTED BY HASH (c_3_0) BUCKETS 3 properties("replication_num"="3") ;
INSERT INTO t2 VALUES (7, parse_json('{}'), 3, -1069398368, 25921195911359973484727614056435496, ["kux", "BYj"], ''), (3, parse_json('{"onPC8c": ["A", {"DD": null, "2c2Sjw": true, "DG": null, "k7Mz": "z", "I": -1439701340, "lEFJnd": -780663584, "We": null, "JBo4UnpH": 1938690026, "o": "", "Ruf9SBPe1": true, "s": -271196752, "Vk": -17150716, "qbdFqsrI": 0.8635494323241585, "rCG": null, "DmDIA6Uo": 0.7062802388209353}, "", 0.42078714388672867, null, 0.6818179994021628], "C": -201633465, "CX8Qmrrm": true, "Xa5roKo0O": [{"gy8h5V": true, "D": null, "c1U": 0.4079266922683821, "SPDSPEca": true, "m": -611168964, "es": 0.36858186419268746, "lZ": true, "EU": null, "5": true, "f6G0": null, "w": null, "6pmV": true, "2jis": false, "HG": 27113557}, {}, "", {"hUA": "O3r3Rop9j", "YG": false, "o": null}, null, null, {"q": null, "Q": null, "R": true, "Seu": -3.73213493E8, "c": null, "tc9sJwn": 0.12399972454119867, "h": "Vx", "exPT": null, "y": false, "dJKG": true, "pP72": true}, -1.069398368E9, [0.3119689159237231], -284508801, -310062477, true, {"1": false, "uWVkF": true, "OZuKM3z": null, "Tw1pJh59C": 0.9475318728673566}, null, -279608784], "M": ["Q0VU", "", 1938690026, [null, true, 0.9628113737396903, -1069398368, true], {"A": null, "tvi": false, "yth": 0.49271190055111413, "O": 1.938690026E9, "1": 0.9628113737396903, "3": null, "yeP1": 0.49271190055111413, "zP": "y2dghrgP", "oHrhNB": false, "vYMMHBb": true, "caZ": null, "19E": false, "i3Mq": false}, 0.12399972454119867, [-1069398368, "CTv4m03V", true, true, null, "doHlmf"], [true, -1566620648, null, true, null, 0.7665548611560026, null, "kZiPNLL2", -201633465, null, 27113557, true, "J2p", -522991235, null], "W8", [false, null, true], "U"], "m": {"10r": -1439701340, "C": true, "Klj": -373213493, "AgHe6w1j": {"A00": null, "0r": null, "EdGCfAOvV": "fl", "N": 0.975116450703116, "s5tip": null, "jB": 0.662298950335101, "vp": false, "Ec": 0.021228777006102484, "5W1hIER": null, "mJld1": true, "C37jpwH": -201633465, "mL": false, "OX6CS": null}, "ZPqrDWDf": 1909085409, "J": null, "EzFIG9": {"0": null, "q6aO5": "h5WWY", "a3": true, "Ps": -201633465, "HiNg80o": false, "fC7z51v7I": true, "D99g": true, "BCKarG": "NS"}, "b85": [null, null, true, null, false, 0.42078714388672867, false, null, 0.40613260587779243, 1.909085409E9], "t2": [false, true, "FN", true, 1267766354, null, false]}, "n": true, "BPQUnx": true, "T0gx2": false, "7ksgMp7f": "PZ", "584S7pD": true, "5": true, "cy47wIz": [null, false, -1652190586, 1537333178, false, [0.3240419855368689, null, -522991235, null, false], {"AQ": null, "X0fy4RL": "aJOc", "1": -2.01633465E8, "M1": 1267766354, "GTjMV9iv": 0.8554324701294553, "U": -1439701340, "3I": false, "wN": false, "i": null, "Bo": null, "fIBQ": null, "81W": -1848961628}, false, false, "Yh4WK", ["KOEiUtVWh"]], "7": true, "hd": -201633465}'), 2, -1848961628, 58125433747773059659282401751098423, ["", "c", "R6c36aVe", "Zy0qUASF", "", "4GAdV9", "", "MC0nPT4lg"], 'J');
TRUNCATE TABLE t1;
INSERT INTO t3 VALUES ('2010-01-23 08:00:14', true, '2010-01-20 08:00:14', percentile_hash('1182833875'), false, true, 17057098365039402734.999463007, -271196752, percentile_hash('-1432474249'), 2, 0.43117842827037833), ('2010-01-30 08:00:14', true, '2010-01-21 08:00:14', percentile_hash('-611168964'), false, true, 9062348516671674584.519458051, -558893305, percentile_hash('-1069398368'), 0, -2.01633465E8);
INSERT INTO t3 VALUES ('2010-01-05 08:00:14', true, NULL, percentile_hash('-271196752'), false, false, 67594546430697535324.694236180, 1093021412, percentile_hash('110475791'), 13, 0.6485670435969693);
INSERT INTO t0 VALUES ('2010-01-04 08:00:14', '2010-01-29 08:00:14', '2010-01-22 08:00:14', '2010-01-21', -284508801, '2010-01-22 08:00:14', [-780663584, -1652190586, 1174877239, 435646108, -201633465, -1012283463, -868153505, 1569868120, -35377343, 1537333178], 'UvpLB', ["m", "g"], true, 0.00000537763, [225880666, -373213493, -201633465, -1848961628, -1272650775, 435646108, -1628010249, -1628010249, 316656113, 915289717, 1569868120, 2122776035, -279608784, -697496549], 3), ('2010-01-12 08:00:14', '2010-01-09 08:00:14', '2010-01-06 08:00:14', '2010-01-07', -1216856690, '2010-01-20 08:00:14', [-580625275, -1823423203, -2072601590, -1513425165, 1552302656, -1432474249, 650208389, -1125301236, 612973488, 1223149707, -708817857], '', ["d"], true, 0.00000130253, [-611168964, -1499012239, 1345786175, 9223372036854775807, 1684800332, 1066492196, 1182833875, 1787463556, -1747153163, -279608784, 1562565741, -1372491542, -1751070557, 855574579, 1792788954], 9), ('2010-01-01 08:00:14', '2010-01-30 08:00:14', '2010-01-11 08:00:14', '2009-12-31', 1341415454, '2010-01-09 08:00:14', [416141203, 612973488, 2122776035], '', ["lF6T", "Nevby2kk", "QlAAY", "D", "S", "", "SIMP5gpjX"], true, 0.00000632545, [416141203, -1814095438, -1067116515, 1116220203, -780663584, 225880666, 1787463556, -1372157118, 483853059], 29), ('2010-01-25 08:00:14', '2010-01-17 08:00:14', '2010-01-26 08:00:14', '2010-01-09', 1174877239, '2010-01-29 08:00:14', [-611168964, -199770597, 859807446, 324448047, 1509460290, -962143552, 1345786175, -707301799, -1616760814, -2027968157, -1455563731, -17150716, -465324967, 89242532, -611168964], 'rOXVa', ["", "zRJDzxk", "", "", "MKk", "gh", "UbzUO", "MN3vs", "iFjbxL5R"], true, 0.00000143131, [-707301799, -406962886, 316656113, -573508443, 1909085409, -611168964, -1264388687], 18);
INSERT INTO t0 VALUES ('2010-01-07 08:00:14', '2010-01-18 08:00:14', '2010-01-10 08:00:14', '2010-01-01', -3260629, '2010-01-15 08:00:14', [2101475579, 50224047, 1269198156, -1379140848, 27113557], 'lc41XCMG', [], false, 0.00000898258, [1787463556, 1174877239, -783262128, 1174877239, 1684800332, -1823423203, -482214657, -1652190586, -413359154], 12);
INSERT INTO t3 VALUES ('2010-01-28 08:00:14', true, '2010-01-04 08:00:14', percentile_hash('-279608784'), false, true, 84078114362906611317.227678625, -2039252130, percentile_hash('2071040509'), 8, 0.4705283353401115), ('2010-01-03 08:00:14', true, '2010-01-10 08:00:14', percentile_hash('-1272650775'), true, false, 65813234073314028982.406576062, -1549118158, percentile_hash('664623160'), 3, 0.68283732592729);
CREATE VIEW v0 AS SELECT t0_125.c_0_0, t0_125.c_0_1, t0_125.c_0_2, t0_125.c_0_3, t0_125.c_0_4, t0_125.c_0_5, t0_125.c_0_6, t0_125.c_0_7, t0_125.c_0_8, t0_125.c_0_9, t0_125.c_0_10, t0_125.c_0_11, t0_125.c_0_12, t2_127.c_2_0, t2_127.c_2_1, t2_127.c_2_2, t2_127.c_2_3, t2_127.c_2_4, t2_127.c_2_5, t2_127.c_2_6 FROM t0 AS t0_125, t2 AS t2_127 WHERE true ORDER BY t0_125.c_0_7, t2_127.c_2_2, t0_125.c_0_0, t0_125.c_0_2, t0_125.c_0_4, t0_125.c_0_12, t0_125.c_0_10, t2_127.c_2_3, t0_125.c_0_5, t0_125.c_0_7, t0_125.c_0_3, t0_125.c_0_7, t2_127.c_2_0, t0_125.c_0_5, t0_125.c_0_0, t0_125.c_0_12;
INSERT INTO t0 VALUES ('2010-01-16 08:00:14', '2010-01-27 08:00:14', '2010-01-12 08:00:14', '2010-01-10', 518925950, '2010-01-03 08:00:14', [], 'a', ["QK"], false, 0.00000664706, [664623160, 1046488694, -1264388687, -868153505, 431514826, -1410757164, 1085316936, -37089863, 1751361687, -373213493, -611168964, 2140193678, -1566620648], 26), ('2010-01-28 08:00:14', '2010-01-23 08:00:14', '2010-01-02 08:00:14', '2010-01-16', -1163701240, '2010-01-31 08:00:14', [604438632, 2132116655, -610560492, 1477190423], 'dz', ["4d", "", "CVJcdejG", "z8vA3hGq", "E", "MJukg7M", "qezc5t8", "oKg0goC", "UFNkrd"], true, 0.00000224075, [179817823, -248758139, 612973488, -451804404, -2072601590, -257952316], 14);
INSERT INTO t1 VALUES (3, '2010-01-31 08:00:14', 0.00881, '2010-01-30 08:00:14', true, '2010-01-19', '2010-01-21', 12, '2010-01-16 08:00:14', 936.3684822542489, 'p3', '2010-01-17', 6, false, hll_hash('LD'), true, 'wY', 0.35772624415147436), (14, '2010-01-15 08:00:14', 0.00975, '2010-01-27 08:00:14', false, '2010-01-21', '2010-01-20', 0, '2010-01-04 08:00:14', 611.4810875194363, 'yyX', '2010-01-26', 9, true, hll_hash('O'), true, 'uF', 0.6525935247394887);
-- SELECT t3_137.c_3_0 FROM t3 AS t3_137 GROUP BY t3_137.c_3_6, t3_137.c_3_7, t3_137.c_3_0;
-- cardinality: 5
-- SELECT t3_137.c_3_0 FROM t3 AS t3_137 GROUP BY t3_137.c_3_6, t3_137.c_3_7, t3_137.c_3_0 HAVING NULL_OR_EMPTY(NULL) UNION ALL SELECT t3_137.c_3_0 FROM t3 AS t3_137 GROUP BY t3_137.c_3_6, t3_137.c_3_7, t3_137.c_3_0 HAVING (NOT (NULL_OR_EMPTY(NULL) )) UNION ALL SELECT t3_137.c_3_0 FROM t3 AS t3_137 GROUP BY t3_137.c_3_6, t3_137.c_3_7, t3_137.c_3_0 HAVING ((NULL_OR_EMPTY(NULL) ) IS NULL);
-- cardinality: 15


```


```
DROP DATABASE IF EXISTS w3;
CREATE DATABASE w3;
USE w3;
admin set frontend config ("enable_decimal_v3" = "true");
CREATE TABLE t0 (c_0_0 DATETIME NOT NULL ,c_0_1 DATETIME NULL ,c_0_2 DATETIME MAX NOT NULL ,c_0_3 PERCENTILE PERCENTILE_UNION NULL ,c_0_4 DECIMAL(30, 15) MIN NOT NULL ,c_0_5 BIGINT MIN NOT NULL ,c_0_6 BOOLEAN REPLACE NULL ,c_0_7 LARGEINT REPLACE_IF_NOT_NULL NOT NULL ,c_0_8 HLL HLL_UNION NOT NULL ,c_0_9 DOUBLE SUM NULL ,c_0_10 FLOAT SUM NOT NULL ,c_0_11 BOOLEAN REPLACE NULL ) AGGREGATE KEY (c_0_0,c_0_1) PARTITION BY RANGE(c_0_0) (START ("2010-01-01") END ("2022-12-31") EVERY (INTERVAL 118 day)) DISTRIBUTED BY HASH (c_0_1,c_0_0) BUCKETS 3 properties("replication_num"="3") ;
CREATE TABLE IF NOT EXISTS t1 (c_1_0 DATETIME NULL ,c_1_1 VARCHAR(1) NOT NULL ,c_1_2 INT NOT NULL ,c_1_3 DECIMAL(26, 13) NULL ,c_1_4 DATETIME NOT NULL ,c_1_5 SMALLINT NOT NULL ,c_1_6 CHAR(21) NOT NULL ,c_1_7 BOOLEAN NOT NULL ,c_1_8 BIGINT NOT NULL ,c_1_9 BOOLEAN NULL ,c_1_10 ARRAY NOT NULL ,c_1_11 LARGEINT NOT NULL ,c_1_12 ARRAY NULL ) UNIQUE KEY (c_1_0,c_1_1,c_1_2,c_1_3,c_1_4,c_1_5,c_1_6) PARTITION BY RANGE(c_1_0) (START ("2010-01-01") END ("2022-12-31") EVERY (INTERVAL 96 day)) DISTRIBUTED BY HASH (c_1_0,c_1_6) BUCKETS 3 properties("replication_num"="3") ;
INSERT INTO t1 VALUES ('2010-01-12 08:00:14', '', 5, 0.8202047314867, '2010-01-24 08:00:14', 7, 'frkJ', true, -63145600, false, ["ENQNkSka0", "589i", "Xn", "AZQg", "YH", "SYlTVa"], -63145600, ["it4VXWx", "MzLXdH1lT", "", "9", "", "cb", "fL4", "GPNOA2CW", "S", "", "j", "y4TMAcuK"]);
INSERT INTO t1 VALUES ('2010-01-05 08:00:14', '', 12, 0.3883669079902, '2010-01-03 08:00:14', 3, '', true, -63145600, NULL, ["j", "", "c", "oqUkI"], -1683869116, ["zMCatIAr", "", "xI", "Wi3Q6k8", "uGH"]), ('2010-01-22 08:00:14', '', 28, NULL, '2010-01-05 08:00:14', 9, 'l', false, 272722550, true, ["ysBmeqciE", "0d", "", "slvr1K6", "rj"], 272722550, ["m", "4r8", "4ZtV", "eQ", "NN", "SlXkuQxeu", "RvQ9BCPS", "yr8tfl7", "Uz", "x", "", "Ps", "", ""]);
INSERT INTO t1 VALUES ('2010-01-15 08:00:14', '', 17, NULL, '2010-01-19 08:00:14', 3, 'kjYIWb', false, -540601585, false, ["a", "i2F", ""], -63145600, ["nBdr5Z4v", "N", "", "1HLA", ""]), (NULL, '', 14, 0.6618294043950, '2010-01-06 08:00:14', 0, 'pEy', true, -1569593817, NULL, ["OXp", "3", "", "", "", "SBqcCac", "UJV", "2", "", "Tpf", "1MU0", "", "df03Nr", "j0"], -1598032962, ["", "3Hnd", "ooa", "", "1", "", "2AL", "x2Z", "o7zo"]);
INSERT INTO t0 VALUES ('2010-01-18 08:00:14', '2010-01-01 08:00:14', '2010-01-28 08:00:14', percentile_hash('-1280109443'), 0.792545939778886, -540601585, false, 759402767, hll_hash('E'), 0.36578366904164006, -1.569593817E9, false), ('2010-01-16 08:00:14', '2010-01-20 08:00:14', '2010-01-15 08:00:14', percentile_hash('-1331026276'), 0.174210370377591, -540601585, true, 1383880049, hll_hash('9w5ZCQ'), 0.1024078292346925, 0.501470743231944, true);
INSERT INTO t1 VALUES ('2010-01-22 08:00:14', '', 5, 0.7812035922271, '2010-01-21 08:00:14', 10, 'OTrR', true, -1569593817, true, ["1", "I", "GJ", "ol19qzfJ", "", "F5ar8", "", "yaa", "r", "6", "hUcdLpF"], -1683869116, ["P", "R", "", "", "Yb3", "n2i", "O", "Fo20", "", "", "wgO"]), ('2010-01-13 08:00:14', '', 5, 0.2602593868005, '2010-01-08 08:00:14', 4, '2', true, -442036666, NULL, ["yY", "lH3Rt", "HuQolt6eS", "xeLAvvs", "", "U", "", "22SkL9K8"], 1690279776, ["", "Y5me", "P", "eCYx"]);
INSERT INTO t1 VALUES ('2010-01-26 08:00:14', '', 12, NULL, '2010-01-27 08:00:14', 13, 'Ab', true, 1249208943, true, ["L", "w", "eVP7aAc", "", "", "ytAtbZ", ""], 759402767, ["5Br", "nM4sS", "", "T", "TE", "", "iUn", "D73T", "3y", "WsXKUcYP", "A0CXczD", "g", "Eu", "6dsaT"]);
INSERT INTO t0 VALUES ('2010-01-02 08:00:14', '2010-01-14 08:00:14', '2010-01-23 08:00:14', percentile_hash('1818859600'), 0.298206509395304, 759402767, NULL, NULL, hll_hash(''), 2.7272255E8, 0.046802108302694156, false), ('2010-01-28 08:00:14', '2010-01-09 08:00:14', '2010-01-19 08:00:14', percentile_hash('1100241214'), 0.454444729356550, -872123264, NULL, 1416302489, hll_hash('P32geI'), 0.0601928213984263, 1.690279776E9, NULL), ('2009-12-31 08:00:14', '2010-01-15 08:00:14', '2010-01-10 08:00:14', percentile_hash('-1683869116'), 0.180945330093009, -442036666, false, 157584458, hll_hash('0'), -1.331026276E9, -1.280109443E9, true);
INSERT INTO t1 VALUES ('2010-01-21 08:00:14', '', 4, 0.1979235019717, '2010-01-24 08:00:14', 12, 'tx1Gb0y2B', false, 1973356944, false, ["NKBtcsPi", "A", "6uoCU", "34", "7", "HC", "", "ZM4m", "D4ydOSx9j", "DShpd"], -542577390, ["qC", "W", "p1VR", "l", "E", "9v3TGdDWY", "yT6", "3pPF9", "Vz", "hSNBoBbjJ", "js4", "9vfUZ0", "zoOO", "8", "erd"]);
INSERT INTO t1 VALUES ('2010-01-25 08:00:14', '', 22, 0.2186325307534, '2010-01-04 08:00:14', 12, 'V0nMYh6F', false, 172802601, NULL, ["uZivK", "24", "yGIKuSw", "wx5M6Tg2G", ""], 2050745680, ["eC", "J", "qSUlpYTDx", "29", "FdWPm8e", "k5", "O", "K0LKQv0Ip", "Js", "j", "sR0SqsD", "", "1dlr"]);
INSERT INTO t0 VALUES ('2010-01-10 08:00:14', '2010-01-09 08:00:14', '2010-01-22 08:00:14', percentile_hash('172802601'), 0.076080091472381, 878223015, true, -1194988227, hll_hash(''), 0.0601928213984263, 0.7728148541072585, true);
INSERT INTO t1 VALUES ('2010-01-22 08:00:14', '', 5, 0.6008561713111, '2010-01-15 08:00:14', 5, '', true, 1455247116, true, ["G", "NzMc", "d5", "Dh", "IFF1omr", "", "ulC", "DWgKlVZxN", "k", "g", "Wwm5g", "a", "FH8W8QpQ6", "FWhZlMx3A"], -1194988227, ["M", "a", "9o", "", "V", "Ua", "Bk", "dLM", "8", "5vjZKz", "rAx", "", "", "vvmiGJH", ""]), ('2010-01-27 08:00:14', '', 27, 0.8182266496809, '2010-01-27 08:00:14', 14, '', true, 110278149, true, ["w1xCTj", "s", "", "s", "hpvi", "B3e6aee0", "j", "T", "", ""], 689936510, ["aYc6", "", "rUYkO", "N", "q", "P", "8kzX8aJj6"]);
INSERT INTO t1 VALUES ('2010-01-24 08:00:14', '', 29, 0.2237429052483, '2010-01-15 08:00:14', 11, 'F', true, -664929238, false, [""], -623276025, ["", "jl9Fwl", "Xy", "ep3XWFt", "VQe", "", "", "", "7erViW", "e", "w", "P", "U1iUu0"]), ('2010-01-19 08:00:14', '', 16, 0.5167410411685, '2010-01-01 08:00:14', 11, 'dpAW2lR', false, -2105129791, true, [], 1703681338, ["3", "ln5", "4G", "E", "", "", "", "XEh", "o", "i", "N", "Dzw26oDM"]), ('2010-01-22 08:00:14', '', 24, 0.9178263900066, '2010-01-26 08:00:14', 3, '', true, -1260011116, false, ["YvjL81fR", "4ksR", "fK", "NNqutijP", "", "", "tE"], 110278149, ["", ""]);
CREATE VIEW v0 AS SELECT t1_50.c_1_0, t1_50.c_1_1, t1_50.c_1_2, t1_50.c_1_3, t1_50.c_1_4, t1_50.c_1_5, t1_50.c_1_6, t1_50.c_1_7, t1_50.c_1_8, t1_50.c_1_9, t1_50.c_1_11 FROM t1 AS t1_50 WHERE (NOT (false)) GROUP BY t1_50.c_1_0, t1_50.c_1_1, t1_50.c_1_2, t1_50.c_1_3, t1_50.c_1_4, t1_50.c_1_5, t1_50.c_1_6, t1_50.c_1_7, t1_50.c_1_8, t1_50.c_1_9, t1_50.c_1_11 ORDER BY t1_50.c_1_4, t1_50.c_1_4, t1_50.c_1_11, t1_50.c_1_6, t1_50.c_1_7, t1_50.c_1_3, t1_50.c_1_3, t1_50.c_1_11, t1_50.c_1_8, t1_50.c_1_1;
INSERT INTO t0 VALUES ('2010-01-24 08:00:14', '2010-01-09 08:00:14', '2010-01-26 08:00:14', percentile_hash('-383622571'), 0.479805037760493, 1368484479, false, 723713955, hll_hash(''), 0.5987237936941517, 1.249208943E9, false), ('2010-01-04 08:00:14', '2010-01-14 08:00:14', '2010-01-23 08:00:14', percentile_hash('-532917514'), 0.828062793704816, 759402767, NULL, NULL, hll_hash('0C'), 0.002459879789817987, -1.280109443E9, false);
INSERT INTO t1 VALUES ('2010-01-29 08:00:14', '', 31, 0.5100736301054, '2010-01-03 08:00:14', 3, 'Q', true, -54834903, true, ["y", "v9vb", "gj"], 1055532875, ["6N0", "U5ig", "WRuFzkuCS", "gNzT", "O4", "tGm", "6x", "WS0MQ", "", "S9p", "5", "l", "oP", "g", "u1lOQU6"]);
INSERT INTO t0 VALUES ('2010-01-20 08:00:14', '2010-01-15 08:00:14', '2010-01-16 08:00:14', percentile_hash('172802601'), 0.502486402371113, 359785339, false, 110278149, hll_hash('a3'), NULL, 0.4531095051745787, false), ('2010-01-23 08:00:14', '2010-01-20 08:00:14', '2010-01-25 08:00:14', percentile_hash('166037822'), 0.829193748521890, 1383880049, true, 1503088321, hll_hash('o'), 0.5915825230806343, -6.64929238E8, false);
INSERT INTO t0 VALUES ('2010-01-03 08:00:14', NULL, '2010-01-26 08:00:14', percentile_hash('449189335'), 0.832924277486566, 759402767, true, -664929238, hll_hash('37DU1hXx'), -8.72123264E8, 0.3274199923857031, false), ('2010-01-10 08:00:14', '2010-01-21 08:00:14', '2010-01-14 08:00:14', percentile_hash('435697369'), 0.671843269495538, -623276025, true, 2076343173, hll_hash(''), 7.59402767E8, 0.6397474342226216, true);
INSERT INTO t1 VALUES ('2010-01-27 08:00:14', '', 19, 0.5139492789708, '2010-01-06 08:00:14', 5, 'AK', true, 878223015, true, ["5Nq4g4g", "gfdu", "", "", "Lp3H", "", "", "", "H", "", "xgyK6lwX"], 2064812976, ["lp", "", "PzQBX", "C", "B", "ZTyhy1alh", "aZE"]);
INSERT INTO t0 VALUES ('2009-12-31 08:00:14', NULL, '2010-01-05 08:00:14', percentile_hash('-113847854'), 0.605118036855461, 388532945, false, -615782405, hll_hash(''), 0.9722877260139795, 0.501470743231944, true);
INSERT INTO t1 VALUES ('2010-01-23 08:00:14', '', 25, 0.0651966586455, '2010-01-06 08:00:14', 9, 'R', true, 1982967937, false, ["mEG", "", "v", "", "P", "owG6", "Q", "", "", ""], 2010755685, ["m", "", "9Z7", "9V", "xnsKIM", "", "RasPEIWy", "0", "OzZi", "qqxaN", "koKY3", "ScYhiPaB"]), ('2009-12-31 08:00:14', '', 20, 0.8022973773215, '2010-01-27 08:00:14', 12, 'S', true, -468845387, NULL, ["i", "", "d99", "", "XJK1hk", "h", "p", "Z", "rt", "b", "", "PM7CCw", "b", "2qh", "5p"], -383622571, ["", "oBi", "T", "", "Qbm", "", "", "v", "Yep", "", "mZ", "zx4LX6m", "tdWuS", "", "8G"]), ('2010-01-24 08:00:14', '', 29, 0.3045631604316, '2010-01-02 08:00:14', 12, '', true, 999241393, true, ["NOdqERCS6", "Jqal", "p69", "f0Y", "", "Zy", "", "PTT", "Kx"], -383622571, ["", "", "xq", "T", "Vdl2kITP", "bVW", "k", "", "aP", "p0CL", "Io", "9m", "", "n2f"]);
INSERT INTO t0 VALUES ('2009-12-31 08:00:14', '2010-01-16 08:00:14', '2010-01-12 08:00:14', percentile_hash('-383622571'), 0.012067007330392, -1783457709, false, 1973356944, hll_hash('MFdoe7t'), 0.9990004240220068, 0.5890124913634565, true), ('2010-01-03 08:00:14', '2010-01-21 08:00:14', '2010-01-04 08:00:14', percentile_hash('-123457164'), 0.094034281484676, 1388512516, true, -1713942689, hll_hash('Pps'), 0.3274199923857031, 0.7097373743307518, true), ('2010-01-06 08:00:14', '2010-01-07 08:00:14', '2010-01-08 08:00:14', percentile_hash('293767322'), 0.486582685988883, 55698628, true, -2078861294, hll_hash(''), 0.6053362985683556, 0.29968018109364136, true);
INSERT INTO t0 VALUES ('2010-01-22 08:00:14', '2010-01-02 08:00:14', '2010-01-04 08:00:14', percentile_hash('60088723'), 0.525883653276312, -492235240, true, 2076343173, hll_hash('Zk8I'), 0.20663870702849152, 0.14720372545544358, false), ('2010-01-11 08:00:14', '2010-01-06 08:00:14', '2010-01-07 08:00:14', percentile_hash('-999670000'), 0.751361071634438, 411005371, false, 1085849381, hll_hash(''), 0.47923104041911146, 1.085849381E9, true);
-- SELECT '1969-12-25 01:41:47' FROM (SELECT v0_56.c_1_0, v0_56.c_1_1, v0_56.c_1_2, v0_56.c_1_3, v0_56.c_1_4, v0_56.c_1_5, v0_56.c_1_6, v0_56.c_1_7, v0_56.c_1_8, v0_56.c_1_9, v0_56.c_1_11 FROM v0 AS v0_56) subv0 LEFT ANTI JOIN t0 AS t0_54 ON subv0.c_1_5 = t0_54.c_0_7 AND ((-1398522340)(-269827043)) >= ((NULL)/(435697369)) AND subv0.c_1_11 = t0_54.c_0_7 AND subv0.c_1_11 = t0_54.c_0_7;
-- cardinality: 15
-- SELECT '1969-12-25 01:41:47' FROM (SELECT v0_56.c_1_0, v0_56.c_1_1, v0_56.c_1_2, v0_56.c_1_3, v0_56.c_1_4, v0_56.c_1_5, v0_56.c_1_6, v0_56.c_1_7, v0_56.c_1_8, v0_56.c_1_9, v0_56.c_1_11 FROM v0 AS v0_56) subv0 LEFT ANTI JOIN t0 AS t0_54 ON subv0.c_1_5 = t0_54.c_0_7 AND ((-1398522340)(-269827043)) >= ((NULL)/(435697369)) AND subv0.c_1_11 = t0_54.c_0_7 AND subv0.c_1_11 = t0_54.c_0_7 WHERE (NOT ((NOT (CAST(((872459142)%(((-420316705)(-1683869116)))) AS BOOLEAN ) )))) UNION ALL SELECT '1969-12-25 01:41:47' FROM (SELECT v0_56.c_1_0, v0_56.c_1_1, v0_56.c_1_2, v0_56.c_1_3, v0_56.c_1_4, v0_56.c_1_5, v0_56.c_1_6, v0_56.c_1_7, v0_56.c_1_8, v0_56.c_1_9, v0_56.c_1_11 FROM v0 AS v0_56) subv0 LEFT ANTI JOIN t0 AS t0_54 ON subv0.c_1_5 = t0_54.c_0_7 AND ((-1398522340)(-269827043)) >= ((NULL)/(435697369)) AND subv0.c_1_11 = t0_54.c_0_7 AND subv0.c_1_11 = t0_54.c_0_7 WHERE (NOT ((NOT ((NOT (CAST(((872459142)%(((-420316705)(-1683869116)))) AS BOOLEAN ) )))))) UNION ALL SELECT '1969-12-25 01:41:47' FROM (SELECT v0_56.c_1_0, v0_56.c_1_1, v0_56.c_1_2, v0_56.c_1_3, v0_56.c_1_4, v0_56.c_1_5, v0_56.c_1_6, v0_56.c_1_7, v0_56.c_1_8, v0_56.c_1_9, v0_56.c_1_11 FROM v0 AS v0_56) subv0 LEFT ANTI JOIN t0 AS t0_54 ON subv0.c_1_5 = t0_54.c_0_7 AND ((-1398522340)(-269827043)) >= ((NULL)/(435697369)) AND subv0.c_1_11 = t0_54.c_0_7 AND subv0.c_1_11 = t0_54.c_0_7 WHERE (((NOT ((NOT (CAST(((872459142)%(((-420316705)(-1683869116)))) AS BOOLEAN ) ))))) IS NULL);
-- cardinality: 45

```

```
DROP DATABASE IF EXISTS h5;
CREATE DATABASE h5;
USE h5;
admin set frontend config ("enable_decimal_v3" = "true");
CREATE TABLE t0 (c_0_0 DATE NULL ,c_0_1 DECIMAL(30, 10) NULL ,c_0_2 DATE NOT NULL ,c_0_3 DECIMAL(15, 5) NOT NULL ,c_0_4 BOOLEAN NOT NULL ,c_0_5 CHAR(21) REPLACE_IF_NOT_NULL NOT NULL ,c_0_6 BOOLEAN REPLACE NOT NULL ,c_0_7 BOOLEAN REPLACE NULL ,c_0_8 FLOAT MAX NULL ,c_0_9 HLL HLL_UNION NOT NULL ,c_0_10 DECIMAL(28, 2) MIN NOT NULL , INDEX index1_c_0_2(c_0_2) USING BITMAP) AGGREGATE KEY (c_0_0,c_0_1,c_0_2,c_0_3,c_0_4) PARTITION BY RANGE(c_0_2) (START ("2010-01-01") END ("2022-12-31") EVERY (INTERVAL 339 day)) DISTRIBUTED BY HASH (c_0_3) BUCKETS 3 properties("replication_num"="3") ;
CREATE TABLE t1 (c_1_0 BOOLEAN NULL ,c_1_1 SMALLINT SUM NULL ,c_1_2 FLOAT MIN NULL ,c_1_3 SMALLINT REPLACE_IF_NOT_NULL NOT NULL ,c_1_4 DECIMAL(31, 9) SUM NOT NULL ,c_1_5 PERCENTILE PERCENTILE_UNION NULL ,c_1_6 VARCHAR(1) MIN NULL ) AGGREGATE KEY (c_1_0) DISTRIBUTED BY HASH (c_1_0) BUCKETS 3 properties("replication_num"="3") ;
CREATE TABLE IF NOT EXISTS t1 (c_1_0 VARCHAR(21) NOT NULL,c_1_1 BOOLEAN NOT NULL,c_1_2 DATE NOT NULL,c_1_3 BOOLEAN NOT NULL,c_1_4 DECIMAL(12, 7) NOT NULL,c_1_5 DATE NOT NULL,c_1_6 BOOLEAN NOT NULL,c_1_7 ARRAY NOT NULL) PRIMARY KEY (c_1_0,c_1_1,c_1_2,c_1_3) DISTRIBUTED BY HASH (c_1_2,c_1_0) BUCKETS 3 properties("replication_num"="3") ;
CREATE TABLE t2 (c_2_0 DATETIME NOT NULL,c_2_1 DATE NOT NULL,c_2_2 VARCHAR(11) NOT NULL,c_2_3 VARCHAR(1) NOT NULL,c_2_4 DATE NOT NULL,c_2_5 DATETIME NOT NULL,c_2_6 DATETIME NOT NULL,c_2_7 INT NOT NULL,c_2_8 VARCHAR(1) NOT NULL,c_2_9 BIGINT NOT NULL,c_2_10 ARRAY NOT NULL,c_2_11 DOUBLE NOT NULL,c_2_12 ARRAY NOT NULL,c_2_13 DECIMAL(12, 11) NOT NULL,c_2_14 DECIMAL(9, 2) NOT NULL,c_2_15 ARRAY NOT NULL,c_2_16 ARRAY NOT NULL,c_2_17 ARRAY NOT NULL, INDEX index1_c_2_0(c_2_0) USING BITMAP) PRIMARY KEY (c_2_0,c_2_1,c_2_2,c_2_3,c_2_4,c_2_5,c_2_6,c_2_7,c_2_8) PARTITION BY RANGE(c_2_1) (START ("2010-01-01") END ("2022-12-31") EVERY (INTERVAL 135 day)) DISTRIBUTED BY HASH (c_2_8,c_2_5,c_2_2,c_2_1,c_2_6,c_2_3,c_2_4,c_2_7,c_2_0) BUCKETS 3 properties("replication_num"="3", "bloom_filter_columns"="c_2_8") ;
TRUNCATE TABLE t0;
DELETE FROM t0 WHERE c_0_2='1969-12-08' AND c_0_4=true;
INSERT INTO t0 VALUES ('2010-01-06', 2516048811.0674870575, '2010-01-11', 22272.37835, false, NULL, false, false, NULL, hll_hash('HczwcITu'), 939270949672947006447998.31), ('2010-01-23', 3716649881.9267906468, '2010-01-17', 69025.75415, true, '', true, false, NULL, hll_hash('rSg'), 298950611755034012517739.94);
INSERT INTO t2 VALUES ('2010-01-14 08:00:14', '2010-01-31', 'h', '', '2010-01-10', '2010-01-01 08:00:14', '2010-01-20 08:00:14', 1, '', 996874355, ["Kv", "iJeDz", "", "g", "dcQAE4Q1"], 9.96874355E8, ["", "", "Q", "tq", "aFyPvSfi", "", "", "", ""], 7E-11, 23253.24, ["vKFUleZXo", "z", "K", ""], [], []);
INSERT INTO t0 VALUES (NULL, 9368968609.0096291485, '2010-01-24', 99747.68287, true, NULL, false, false, 9.96874355E8, hll_hash('PVc'), 323772619848412768206458.19), ('2010-01-30', 2638336845.4241417237, '2010-01-26', 57242.23784, true, NULL, true, NULL, 0.20728488896954445, hll_hash('2b2dLii6'), 159804034799941208078166.78);
INSERT INTO t0 VALUES ('2010-01-12', 3267247884.4305137034, '2010-01-29', 92363.64923, true, '', true, false, 0.04982029037278557, hll_hash('Qd'), 164522478044373920980092.47), (NULL, 2905265066.0352713818, '2010-01-26', 81754.15619, true, '', false, false, 0.04982029037278557, hll_hash('jqi'), 518427844928774694821527.21);
-- SELECT t0_56.c_0_0 FROM t0 AS t0_56, t1 AS t1_57 GROUP BY t0_56.c_0_2, t0_56.c_0_0;
-- cardinality: 0
-- SELECT t0_56.c_0_0 FROM t0 AS t0_56, t1 AS t1_57 GROUP BY t0_56.c_0_2, t0_56.c_0_0 HAVING MIN( ( t1_57.c_1_2 ) >= ( t1_57.c_1_2 ) ) UNION ALL SELECT t0_56.c_0_0 FROM t0 AS t0_56, t1 AS t1_57 GROUP BY t0_56.c_0_2, t0_56.c_0_0 HAVING (NOT (MIN( ( t1_57.c_1_2 ) >= ( t1_57.c_1_2 ) ) )) UNION ALL SELECT t0_56.c_0_0 FROM t0 AS t0_56, t1 AS t1_57 GROUP BY t0_56.c_0_2, t0_56.c_0_0 HAVING ((MIN( ( t1_57.c_1_2 ) >= ( t1_57.c_1_2 ) ) ) IS NULL);
-- cardinality: 6


```
### StarRocks version (Required)
 - You can get the StarRocks version by executing SQL `select current_version()`
 - +----------------------+
| ；                   |
+----------------------+
| MAIN-RELEASE 58f7979 |
+----------------------+
1 row in set (0.06 sec)

Error Message
--java.lang.AssertionError: the size of the result sets mismatch (5 and 15)!
---- SELECT t3_137.c_3_0 FROM t3 AS t3_137 GROUP BY t3_137.c_3_6, t3_137.c_3_7, t3_137.c_3_0;
---- cardinality: 5
---- SELECT t3_137.c_3_0 FROM t3 AS t3_137 GROUP BY t3_137.c_3_6, t3_137.c_3_7, t3_137.c_3_0 HAVING NULL_OR_EMPTY(NULL)  UNION ALL SELECT t3_137.c_3_0 FROM t3 AS t3_137 GROUP BY t3_137.c_3_6, t3_137.c_3_7, t3_137.c_3_0 HAVING (NOT (NULL_OR_EMPTY(NULL) )) UNION ALL SELECT t3_137.c_3_0 FROM t3 AS t3_137 GROUP BY t3_137.c_3_6, t3_137.c_3_7, t3_137.c_3_0 HAVING ((NULL_OR_EMPTY(NULL) ) IS NULL);
---- cardinality: 15
--	at sqlancer.ComparatorHelper.assumeResultSetsAreEqual(ComparatorHelper.java:104)
--	at sqlancer.starrocks.oracle.tlp.StarRocksTLPHavingOracle.check(StarRocksTLPHavingOracle.java:56)
--	at sqlancer.ProviderAdapter.generateAndTestDatabase(ProviderAdapter.java:55)
--	at sqlancer.Main$DBMSExecutor.run(Main.java:326)
--	at sqlancer.Main$2.run(Main.java:521)
--	at sqlancer.Main$2.runThread(Main.java:498)
--	at sqlancer.Main$2.run(Main.java:488)
--	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149)
--	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624)
--	at java.lang.Thread.run(Thread.java:748)
---- Time: 2022/10/11 23:18:47
-- Database: h9
-- Database version: 5.1.0
-- seed value: 1665500741604
DROP DATABASE IF EXISTS h9;
CREATE DATABASE h9;
USE h9;
admin set frontend config ("enable_decimal_v3" = "true");
CREATE TABLE IF NOT EXISTS t0 (c_0_0 DATETIME NOT NULL,c_0_1 DATETIME NOT NULL,c_0_2 DATETIME NOT NULL,c_0_3 DATE NOT NULL,c_0_4 LARGEINT NOT NULL,c_0_5 DATETIME NOT NULL,c_0_6 ARRAY<INT> NOT NULL,c_0_7 VARCHAR(31) NOT NULL,c_0_8 ARRAY<STRING> NOT NULL,c_0_9 BOOLEAN NOT NULL,c_0_10 DECIMAL(17, 11)  NOT NULL,c_0_11 ARRAY<INT> NOT NULL,c_0_12 INT NOT NULL, INDEX index1_c_0_2(`c_0_2`) USING BITMAP) PRIMARY KEY (c_0_0,c_0_1,c_0_2,c_0_3,c_0_4) PARTITION BY RANGE(c_0_0) (START ("2010-01-01") END ("2022-12-31") EVERY (INTERVAL 1186 day)) DISTRIBUTED BY HASH (c_0_0) BUCKETS 3 properties("replication_num"="3") ;
CREATE TABLE t1 (c_1_0 SMALLINT NOT NULL ,c_1_1 DATETIME NOT NULL ,c_1_2 DECIMAL(8, 5)  NOT NULL ,c_1_3 DATETIME NULL ,c_1_4 BOOLEAN NOT NULL ,c_1_5 DATE NULL ,c_1_6 DATE NOT NULL ,c_1_7 SMALLINT NOT NULL ,c_1_8 DATETIME NOT NULL ,c_1_9 DECIMAL(29, 13)  NULL ,c_1_10 CHAR(11) NULL ,c_1_11 DATE MAX NOT NULL ,c_1_12 SMALLINT MIN NULL ,c_1_13 BOOLEAN REPLACE NULL ,c_1_14 HLL HLL_UNION NOT NULL ,c_1_15 BOOLEAN REPLACE NOT NULL ,c_1_16 VARCHAR(11) MIN NOT NULL ,c_1_17 DOUBLE REPLACE NULL , INDEX index1_c_1_3(`c_1_3`) USING BITMAP, INDEX index1_c_1_8(`c_1_8`) USING BITMAP) AGGREGATE KEY (c_1_0,c_1_1,c_1_2,c_1_3,c_1_4,c_1_5,c_1_6,c_1_7,c_1_8,c_1_9,c_1_10) PARTITION BY RANGE(c_1_5) (START ("2010-01-01") END ("2022-12-31") EVERY (INTERVAL 110 day)) DISTRIBUTED BY HASH (c_1_2,c_1_3,c_1_8,c_1_7) BUCKETS 3 properties("replication_num"="3") ;
CREATE TABLE IF NOT EXISTS t2 (c_2_0 TINYINT NULL ,c_2_1 JSON NULL ,c_2_2 TINYINT NULL ,c_2_3 LARGEINT NOT NULL ,c_2_4 DECIMAL(35, 0)  NOT NULL ,c_2_5 ARRAY<STRING> NULL ,c_2_6 VARCHAR(11) NOT NULL ) DUPLICATE KEY (c_2_0) DISTRIBUTED BY HASH (c_2_0) BUCKETS 3 properties("replication_num"="3") ;
CREATE TABLE t3 (c_3_0 DATETIME NOT NULL ,c_3_1 BOOLEAN REPLACE NOT NULL ,c_3_2 DATETIME REPLACE_IF_NOT_NULL NULL ,c_3_3 PERCENTILE PERCENTILE_UNION NOT NULL ,c_3_4 BOOLEAN REPLACE NOT NULL ,c_3_5 BOOLEAN REPLACE NOT NULL ,c_3_6 DECIMAL(24, 9)  SUM NOT NULL ,c_3_7 BIGINT MAX NOT NULL ,c_3_8 PERCENTILE PERCENTILE_UNION NOT NULL ,c_3_9 SMALLINT REPLACE NOT NULL ,c_3_10 FLOAT MIN NOT NULL ) AGGREGATE KEY (c_3_0) PARTITION BY RANGE(c_3_0) (START ("2010-01-01") END ("2022-12-31") EVERY (INTERVAL 96 day)) DISTRIBUTED BY HASH (c_3_0) BUCKETS 3 properties("replication_num"="3") ;
INSERT INTO t2 VALUES (7, parse_json('{}'), 3, -1069398368, 25921195911359973484727614056435496, ["kux", "BYj"], ''), (3, parse_json('{"onPC8c": ["A", {"DD": null, "2c2Sjw": true, "DG": null, "k7Mz": "z", "I": -1439701340, "lEFJnd": -780663584, "We": null, "JBo4UnpH": 1938690026, "o": "", "Ruf9SBPe1": true, "s": -271196752, "Vk": -17150716, "qbdFqsrI": 0.8635494323241585, "rCG": null, "DmDIA6Uo": 0.7062802388209353}, "", 0.42078714388672867, null, 0.6818179994021628], "C": -201633465, "CX8Qmrrm": true, "Xa5roKo0O": [{"gy8h5V": true, "D": null, "c1U": 0.4079266922683821, "SPDSPEca": true, "m": -611168964, "es": 0.36858186419268746, "lZ": true, "EU": null, "5": true, "f6G0": null, "w": null, "6pmV": true, "2jis": false, "HG": 27113557}, {}, "", {"hUA": "O3r3Rop9j", "YG": false, "o": null}, null, null, {"q": null, "Q": null, "R": true, "Seu": -3.73213493E8, "c": null, "tc9sJwn": 0.12399972454119867, "h": "Vx", "exPT": null, "y": false, "dJKG": true, "pP72": true}, -1.069398368E9, [0.3119689159237231], -284508801, -310062477, true, {"1": false, "uWVkF": true, "OZuKM3z": null, "Tw1pJh59C": 0.9475318728673566}, null, -279608784], "M": ["Q0VU", "", 1938690026, [null, true, 0.9628113737396903, -1069398368, true], {"A": null, "tvi": false, "yth": 0.49271190055111413, "O": 1.938690026E9, "1": 0.9628113737396903, "3": null, "yeP1": 0.49271190055111413, "zP": "y2dghrgP", "oHrhNB": false, "vYMMHBb": true, "caZ": null, "19E": false, "i3Mq": false}, 0.12399972454119867, [-1069398368, "CTv4m03V", true, true, null, "doHlmf"], [true, -1566620648, null, true, null, 0.7665548611560026, null, "kZiPNLL2", -201633465, null, 27113557, true, "J2p", -522991235, null], "W8", [false, null, true], "U"], "m": {"10r": -1439701340, "C": true, "Klj": -373213493, "AgHe6w1j": {"A00": null, "0r": null, "EdGCfAOvV": "fl", "N": 0.975116450703116, "s5tip": null, "jB": 0.662298950335101, "vp": false, "Ec": 0.021228777006102484, "5W1hIER": null, "mJld1": true, "C37jpwH": -201633465, "mL": false, "OX6CS": null}, "ZPqrDWDf": 1909085409, "J": null, "EzFIG9": {"0": null, "q6aO5": "h5WWY", "a3": true, "Ps": -201633465, "HiNg80o": false, "fC7z51v7I": true, "D99g": true, "BCKarG": "NS"}, "b85": [null, null, true, null, false, 0.42078714388672867, false, null, 0.40613260587779243, 1.909085409E9], "t2": [false, true, "FN", true, 1267766354, null, false]}, "n": true, "BPQUnx": true, "T0gx2": false, "7ksgMp7f": "PZ", "584S7pD": true, "5": true, "cy47wIz": [null, false, -1652190586, 1537333178, false, [0.3240419855368689, null, -522991235, null, false], {"AQ": null, "X0fy4RL": "aJOc", "1": -2.01633465E8, "M1": 1267766354, "GTjMV9iv": 0.8554324701294553, "U": -1439701340, "3I": false, "wN": false, "i": null, "Bo": null, "fIBQ": null, "81W": -1848961628}, false, false, "Yh4WK", ["KOEiUtVWh"]], "7": true, "hd": -201633465}'), 2, -1848961628, 58125433747773059659282401751098423, ["", "c", "R6c36aVe", "Zy0qUASF", "", "4GAdV9", "", "MC0nPT4lg"], 'J');
TRUNCATE TABLE t1;
INSERT INTO t3 VALUES ('2010-01-23 08:00:14', true, '2010-01-20 08:00:14', percentile_hash('1182833875'), false, true, 17057098365039402734.999463007, -271196752, percentile_hash('-1432474249'), 2, 0.43117842827037833), ('2010-01-30 08:00:14', true, '2010-01-21 08:00:14', percentile_hash('-611168964'), false, true, 9062348516671674584.519458051, -558893305, percentile_hash('-1069398368'), 0, -2.01633465E8);
INSERT INTO t3 VALUES ('2010-01-05 08:00:14', true, NULL, percentile_hash('-271196752'), false, false, 67594546430697535324.694236180, 1093021412, percentile_hash('110475791'), 13, 0.6485670435969693);
INSERT INTO t0 VALUES ('2010-01-04 08:00:14', '2010-01-29 08:00:14', '2010-01-22 08:00:14', '2010-01-21', -284508801, '2010-01-22 08:00:14', [-780663584, -1652190586, 1174877239, 435646108, -201633465, -1012283463, -868153505, 1569868120, -35377343, 1537333178], 'UvpLB', ["m", "g"], true, 0.00000537763, [225880666, -373213493, -201633465, -1848961628, -1272650775, 435646108, -1628010249, -1628010249, 316656113, 915289717, 1569868120, 2122776035, -279608784, -697496549], 3), ('2010-01-12 08:00:14', '2010-01-09 08:00:14', '2010-01-06 08:00:14', '2010-01-07', -1216856690, '2010-01-20 08:00:14', [-580625275, -1823423203, -2072601590, -1513425165, 1552302656, -1432474249, 650208389, -1125301236, 612973488, 1223149707, -708817857], '', ["d"], true, 0.00000130253, [-611168964, -1499012239, 1345786175, 9223372036854775807, 1684800332, 1066492196, 1182833875, 1787463556, -1747153163, -279608784, 1562565741, -1372491542, -1751070557, 855574579, 1792788954], 9), ('2010-01-01 08:00:14', '2010-01-30 08:00:14', '2010-01-11 08:00:14', '2009-12-31', 1341415454, '2010-01-09 08:00:14', [416141203, 612973488, 2122776035], '', ["lF6T", "Nevby2kk", "QlAAY", "D", "S", "", "SIMP5gpjX"], true, 0.00000632545, [416141203, -1814095438, -1067116515, 1116220203, -780663584, 225880666, 1787463556, -1372157118, 483853059], 29), ('2010-01-25 08:00:14', '2010-01-17 08:00:14', '2010-01-26 08:00:14', '2010-01-09', 1174877239, '2010-01-29 08:00:14', [-611168964, -199770597, 859807446, 324448047, 1509460290, -962143552, 1345786175, -707301799, -1616760814, -2027968157, -1455563731, -17150716, -465324967, 89242532, -611168964], 'rOXVa', ["", "zRJDzxk", "", "", "MKk", "gh", "UbzUO", "MN3vs", "iFjbxL5R"], true, 0.00000143131, [-707301799, -406962886, 316656113, -573508443, 1909085409, -611168964, -1264388687], 18);
INSERT INTO t0 VALUES ('2010-01-07 08:00:14', '2010-01-18 08:00:14', '2010-01-10 08:00:14', '2010-01-01', -3260629, '2010-01-15 08:00:14', [2101475579, 50224047, 1269198156, -1379140848, 27113557], 'lc41XCMG', [], false, 0.00000898258, [1787463556, 1174877239, -783262128, 1174877239, 1684800332, -1823423203, -482214657, -1652190586, -413359154], 12);
INSERT INTO t3 VALUES ('2010-01-28 08:00:14', true, '2010-01-04 08:00:14', percentile_hash('-279608784'), false, true, 84078114362906611317.227678625, -2039252130, percentile_hash('2071040509'), 8, 0.4705283353401115), ('2010-01-03 08:00:14', true, '2010-01-10 08:00:14', percentile_hash('-1272650775'), true, false, 65813234073314028982.406576062, -1549118158, percentile_hash('664623160'), 3, 0.68283732592729);
CREATE VIEW v0 AS SELECT t0_125.c_0_0, t0_125.c_0_1, t0_125.c_0_2, t0_125.c_0_3, t0_125.c_0_4, t0_125.c_0_5, t0_125.c_0_6, t0_125.c_0_7, t0_125.c_0_8, t0_125.c_0_9, t0_125.c_0_10, t0_125.c_0_11, t0_125.c_0_12, t2_127.c_2_0, t2_127.c_2_1, t2_127.c_2_2, t2_127.c_2_3, t2_127.c_2_4, t2_127.c_2_5, t2_127.c_2_6 FROM t0 AS t0_125, t2 AS t2_127 WHERE true ORDER BY t0_125.c_0_7, t2_127.c_2_2, t0_125.c_0_0, t0_125.c_0_2, t0_125.c_0_4, t0_125.c_0_12, t0_125.c_0_10, t2_127.c_2_3, t0_125.c_0_5, t0_125.c_0_7, t0_125.c_0_3, t0_125.c_0_7, t2_127.c_2_0, t0_125.c_0_5, t0_125.c_0_0, t0_125.c_0_12;
INSERT INTO t0 VALUES ('2010-01-16 08:00:14', '2010-01-27 08:00:14', '2010-01-12 08:00:14', '2010-01-10', 518925950, '2010-01-03 08:00:14', [], 'a', ["QK"], false, 0.00000664706, [664623160, 1046488694, -1264388687, -868153505, 431514826, -1410757164, 1085316936, -37089863, 1751361687, -373213493, -611168964, 2140193678, -1566620648], 26), ('2010-01-28 08:00:14', '2010-01-23 08:00:14', '2010-01-02 08:00:14', '2010-01-16', -1163701240, '2010-01-31 08:00:14', [604438632, 2132116655, -610560492, 1477190423], 'dz', ["4d", "", "CVJcdejG", "z8vA3hGq", "E", "MJukg7M", "qezc5t8", "oKg0goC", "UFNkrd"], true, 0.00000224075, [179817823, -248758139, 612973488, -451804404, -2072601590, -257952316], 14);
INSERT INTO t1 VALUES (3, '2010-01-31 08:00:14', 0.00881, '2010-01-30 08:00:14', true, '2010-01-19', '2010-01-21', 12, '2010-01-16 08:00:14', 936.3684822542489, 'p3', '2010-01-17', 6, false, hll_hash('LD'), true, 'wY', 0.35772624415147436), (14, '2010-01-15 08:00:14', 0.00975, '2010-01-27 08:00:14', false, '2010-01-21', '2010-01-20', 0, '2010-01-04 08:00:14', 611.4810875194363, 'yyX', '2010-01-26', 9, true, hll_hash('O'), true, 'uF', 0.6525935247394887);
-- SELECT t3_137.c_3_0 FROM t3 AS t3_137 GROUP BY t3_137.c_3_6, t3_137.c_3_7, t3_137.c_3_0;
-- cardinality: 5
-- SELECT t3_137.c_3_0 FROM t3 AS t3_137 GROUP BY t3_137.c_3_6, t3_137.c_3_7, t3_137.c_3_0 HAVING NULL_OR_EMPTY(NULL)  UNION ALL SELECT t3_137.c_3_0 FROM t3 AS t3_137 GROUP BY t3_137.c_3_6, t3_137.c_3_7, t3_137.c_3_0 HAVING (NOT (NULL_OR_EMPTY(NULL) )) UNION ALL SELECT t3_137.c_3_0 FROM t3 AS t3_137 GROUP BY t3_137.c_3_6, t3_137.c_3_7, t3_137.c_3_0 HAVING ((NULL_OR_EMPTY(NULL) ) IS NULL);
-- cardinality: 15

--java.lang.AssertionError: the size of the result sets mismatch (15 and 45)!
---- SELECT '1969-12-25 01:41:47' FROM  (SELECT v0_56.c_1_0, v0_56.c_1_1, v0_56.c_1_2, v0_56.c_1_3, v0_56.c_1_4, v0_56.c_1_5, v0_56.c_1_6, v0_56.c_1_7, v0_56.c_1_8, v0_56.c_1_9, v0_56.c_1_11 FROM v0 AS v0_56) subv0 LEFT ANTI JOIN t0 AS t0_54 ON subv0.c_1_5 = t0_54.c_0_7 AND ((-1398522340)*(-269827043)) >= ((NULL)/(435697369)) AND subv0.c_1_11 = t0_54.c_0_7 AND subv0.c_1_11 = t0_54.c_0_7;
---- cardinality: 15
---- SELECT '1969-12-25 01:41:47' FROM  (SELECT v0_56.c_1_0, v0_56.c_1_1, v0_56.c_1_2, v0_56.c_1_3, v0_56.c_1_4, v0_56.c_1_5, v0_56.c_1_6, v0_56.c_1_7, v0_56.c_1_8, v0_56.c_1_9, v0_56.c_1_11 FROM v0 AS v0_56) subv0 LEFT ANTI JOIN t0 AS t0_54 ON subv0.c_1_5 = t0_54.c_0_7 AND ((-1398522340)*(-269827043)) >= ((NULL)/(435697369)) AND subv0.c_1_11 = t0_54.c_0_7 AND subv0.c_1_11 = t0_54.c_0_7 WHERE (NOT ((NOT (CAST(((872459142)%(((-420316705)*(-1683869116)))) AS BOOLEAN ) )))) UNION ALL SELECT '1969-12-25 01:41:47' FROM  (SELECT v0_56.c_1_0, v0_56.c_1_1, v0_56.c_1_2, v0_56.c_1_3, v0_56.c_1_4, v0_56.c_1_5, v0_56.c_1_6, v0_56.c_1_7, v0_56.c_1_8, v0_56.c_1_9, v0_56.c_1_11 FROM v0 AS v0_56) subv0 LEFT ANTI JOIN t0 AS t0_54 ON subv0.c_1_5 = t0_54.c_0_7 AND ((-1398522340)*(-269827043)) >= ((NULL)/(435697369)) AND subv0.c_1_11 = t0_54.c_0_7 AND subv0.c_1_11 = t0_54.c_0_7 WHERE (NOT ((NOT ((NOT (CAST(((872459142)%(((-420316705)*(-1683869116)))) AS BOOLEAN ) )))))) UNION ALL SELECT '1969-12-25 01:41:47' FROM  (SELECT v0_56.c_1_0, v0_56.c_1_1, v0_56.c_1_2, v0_56.c_1_3, v0_56.c_1_4, v0_56.c_1_5, v0_56.c_1_6, v0_56.c_1_7, v0_56.c_1_8, v0_56.c_1_9, v0_56.c_1_11 FROM v0 AS v0_56) subv0 LEFT ANTI JOIN t0 AS t0_54 ON subv0.c_1_5 = t0_54.c_0_7 AND ((-1398522340)*(-269827043)) >= ((NULL)/(435697369)) AND subv0.c_1_11 = t0_54.c_0_7 AND subv0.c_1_11 = t0_54.c_0_7 WHERE (((NOT ((NOT (CAST(((872459142)%(((-420316705)*(-1683869116)))) AS BOOLEAN ) ))))) IS NULL);
---- cardinality: 45
--	at sqlancer.ComparatorHelper.assumeResultSetsAreEqual(ComparatorHelper.java:104)
--	at sqlancer.starrocks.oracle.tlp.StarRocksTLPWhereOracle.check(StarRocksTLPWhereOracle.java:50)
--	at sqlancer.ProviderAdapter.generateAndTestDatabase(ProviderAdapter.java:55)
--	at sqlancer.Main$DBMSExecutor.run(Main.java:326)
--	at sqlancer.Main$2.run(Main.java:521)
--	at sqlancer.Main$2.runThread(Main.java:498)
--	at sqlancer.Main$2.run(Main.java:488)
--	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149)
--	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624)
--	at java.lang.Thread.run(Thread.java:748)
---- Time: 2022/10/16 23:00:48
-- Database: w3
-- Database version: 5.1.0
-- seed value: 1665932412899
DROP DATABASE IF EXISTS w3;
CREATE DATABASE w3;
USE w3;
admin set frontend config ("enable_decimal_v3" = "true");
CREATE TABLE t0 (c_0_0 DATETIME NOT NULL ,c_0_1 DATETIME NULL ,c_0_2 DATETIME MAX NOT NULL ,c_0_3 PERCENTILE PERCENTILE_UNION NULL ,c_0_4 DECIMAL(30, 15)  MIN NOT NULL ,c_0_5 BIGINT MIN NOT NULL ,c_0_6 BOOLEAN REPLACE NULL ,c_0_7 LARGEINT REPLACE_IF_NOT_NULL NOT NULL ,c_0_8 HLL HLL_UNION NOT NULL ,c_0_9 DOUBLE SUM NULL ,c_0_10 FLOAT SUM NOT NULL ,c_0_11 BOOLEAN REPLACE NULL ) AGGREGATE KEY (c_0_0,c_0_1) PARTITION BY RANGE(c_0_0) (START ("2010-01-01") END ("2022-12-31") EVERY (INTERVAL 118 day)) DISTRIBUTED BY HASH (c_0_1,c_0_0) BUCKETS 3 properties("replication_num"="3") ;
CREATE TABLE IF NOT EXISTS t1 (c_1_0 DATETIME NULL ,c_1_1 VARCHAR(1) NOT NULL ,c_1_2 INT NOT NULL ,c_1_3 DECIMAL(26, 13)  NULL ,c_1_4 DATETIME NOT NULL ,c_1_5 SMALLINT NOT NULL ,c_1_6 CHAR(21) NOT NULL ,c_1_7 BOOLEAN NOT NULL ,c_1_8 BIGINT NOT NULL ,c_1_9 BOOLEAN NULL ,c_1_10 ARRAY<STRING> NOT NULL ,c_1_11 LARGEINT NOT NULL ,c_1_12 ARRAY<STRING> NULL ) UNIQUE KEY (c_1_0,c_1_1,c_1_2,c_1_3,c_1_4,c_1_5,c_1_6) PARTITION BY RANGE(c_1_0) (START ("2010-01-01") END ("2022-12-31") EVERY (INTERVAL 96 day)) DISTRIBUTED BY HASH (c_1_0,c_1_6) BUCKETS 3 properties("replication_num"="3") ;
INSERT INTO t1 VALUES ('2010-01-12 08:00:14', '', 5, 0.8202047314867, '2010-01-24 08:00:14', 7, 'frkJ', true, -63145600, false, ["ENQNkSka0", "589i", "Xn", "AZQg", "YH", "SYlTVa"], -63145600, ["it4VXWx", "MzLXdH1lT", "", "9", "", "cb", "fL4", "GPNOA2CW", "S", "", "j", "y4TMAcuK"]);
INSERT INTO t1 VALUES ('2010-01-05 08:00:14', '', 12, 0.3883669079902, '2010-01-03 08:00:14', 3, '', true, -63145600, NULL, ["j", "", "c", "oqUkI"], -1683869116, ["zMCatIAr", "", "xI", "Wi3Q6k8", "uGH"]), ('2010-01-22 08:00:14', '', 28, NULL, '2010-01-05 08:00:14', 9, 'l', false, 272722550, true, ["ysBmeqciE", "0d", "", "slvr1K6", "rj"], 272722550, ["m", "4r8", "4ZtV", "eQ", "NN", "SlXkuQxeu", "RvQ9BCPS", "yr8tfl7", "Uz", "x", "", "Ps", "", ""]);
INSERT INTO t1 VALUES ('2010-01-15 08:00:14', '', 17, NULL, '2010-01-19 08:00:14', 3, 'kjYIWb', false, -540601585, false, ["a", "i2F", ""], -63145600, ["nBdr5Z4v", "N", "", "1HLA", ""]), (NULL, '', 14, 0.6618294043950, '2010-01-06 08:00:14', 0, 'pEy', true, -1569593817, NULL, ["OXp", "3", "", "", "", "SBqcCac", "UJV", "2", "", "Tpf", "1MU0", "", "df03Nr", "j0"], -1598032962, ["", "3Hnd", "ooa", "", "1", "", "2AL", "x2Z", "o7zo"]);
INSERT INTO t0 VALUES ('2010-01-18 08:00:14', '2010-01-01 08:00:14', '2010-01-28 08:00:14', percentile_hash('-1280109443'), 0.792545939778886, -540601585, false, 759402767, hll_hash('E'), 0.36578366904164006, -1.569593817E9, false), ('2010-01-16 08:00:14', '2010-01-20 08:00:14', '2010-01-15 08:00:14', percentile_hash('-1331026276'), 0.174210370377591, -540601585, true, 1383880049, hll_hash('9w5ZCQ'), 0.1024078292346925, 0.501470743231944, true);
INSERT INTO t1 VALUES ('2010-01-22 08:00:14', '', 5, 0.7812035922271, '2010-01-21 08:00:14', 10, 'OTrR', true, -1569593817, true, ["1", "I", "GJ", "ol19qzfJ", "", "F5ar8", "", "yaa", "r", "6", "hUcdLpF"], -1683869116, ["P", "R", "", "", "Yb3", "n2i", "O", "Fo20", "", "", "wgO"]), ('2010-01-13 08:00:14', '', 5, 0.2602593868005, '2010-01-08 08:00:14', 4, '2', true, -442036666, NULL, ["yY", "lH3Rt", "HuQolt6eS", "xeLAvvs", "", "U", "", "22SkL9K8"], 1690279776, ["", "Y5me", "P", "eCYx"]);
INSERT INTO t1 VALUES ('2010-01-26 08:00:14', '', 12, NULL, '2010-01-27 08:00:14', 13, 'Ab', true, 1249208943, true, ["L", "w", "eVP7aAc", "", "", "ytAtbZ", ""], 759402767, ["5Br", "nM4sS", "", "T", "TE", "", "iUn", "D73T", "3y", "WsXKUcYP", "A0CXczD", "g", "Eu", "6dsaT"]);
INSERT INTO t0 VALUES ('2010-01-02 08:00:14', '2010-01-14 08:00:14', '2010-01-23 08:00:14', percentile_hash('1818859600'), 0.298206509395304, 759402767, NULL, NULL, hll_hash(''), 2.7272255E8, 0.046802108302694156, false), ('2010-01-28 08:00:14', '2010-01-09 08:00:14', '2010-01-19 08:00:14', percentile_hash('1100241214'), 0.454444729356550, -872123264, NULL, 1416302489, hll_hash('P32geI'), 0.0601928213984263, 1.690279776E9, NULL), ('2009-12-31 08:00:14', '2010-01-15 08:00:14', '2010-01-10 08:00:14', percentile_hash('-1683869116'), 0.180945330093009, -442036666, false, 157584458, hll_hash('0'), -1.331026276E9, -1.280109443E9, true);
INSERT INTO t1 VALUES ('2010-01-21 08:00:14', '', 4, 0.1979235019717, '2010-01-24 08:00:14', 12, 'tx1Gb0y2B', false, 1973356944, false, ["NKBtcsPi", "A", "6uoCU", "34", "7", "HC", "", "ZM4m", "D4ydOSx9j", "DShpd"], -542577390, ["qC", "W", "p1VR", "l", "E", "9v3TGdDWY", "yT6", "3pPF9", "Vz", "hSNBoBbjJ", "js4", "9vfUZ0", "zoOO", "8", "erd"]);
INSERT INTO t1 VALUES ('2010-01-25 08:00:14', '', 22, 0.2186325307534, '2010-01-04 08:00:14', 12, 'V0nMYh6F', false, 172802601, NULL, ["uZivK", "24", "yGIKuSw", "wx5M6Tg2G", ""], 2050745680, ["eC", "J", "qSUlpYTDx", "29", "FdWPm8e", "k5", "O", "K0LKQv0Ip", "Js", "j", "sR0SqsD", "", "1dlr"]);
INSERT INTO t0 VALUES ('2010-01-10 08:00:14', '2010-01-09 08:00:14', '2010-01-22 08:00:14', percentile_hash('172802601'), 0.076080091472381, 878223015, true, -1194988227, hll_hash(''), 0.0601928213984263, 0.7728148541072585, true);
INSERT INTO t1 VALUES ('2010-01-22 08:00:14', '', 5, 0.6008561713111, '2010-01-15 08:00:14', 5, '', true, 1455247116, true, ["G", "NzMc", "d5", "Dh", "IFF1omr", "", "ulC", "DWgKlVZxN", "k", "g", "Wwm5g", "a", "FH8W8QpQ6", "FWhZlMx3A"], -1194988227, ["M", "a", "9o", "", "V", "Ua", "Bk", "dLM", "8", "5vjZKz", "rAx", "", "", "vvmiGJH", ""]), ('2010-01-27 08:00:14', '', 27, 0.8182266496809, '2010-01-27 08:00:14', 14, '', true, 110278149, true, ["w1xCTj", "s", "", "s", "hpvi", "B3e6aee0", "j", "T", "", ""], 689936510, ["aYc6", "", "rUYkO", "N", "q", "P", "8kzX8aJj6"]);
INSERT INTO t1 VALUES ('2010-01-24 08:00:14', '', 29, 0.2237429052483, '2010-01-15 08:00:14', 11, 'F', true, -664929238, false, [""], -623276025, ["", "jl9Fwl", "Xy", "ep3XWFt", "VQe", "", "", "", "7erViW", "e", "w", "P", "U1iUu0"]), ('2010-01-19 08:00:14', '', 16, 0.5167410411685, '2010-01-01 08:00:14', 11, 'dpAW2lR', false, -2105129791, true, [], 1703681338, ["3", "ln5", "4G", "E", "", "", "", "XEh", "o", "i", "N", "Dzw26oDM"]), ('2010-01-22 08:00:14', '', 24, 0.9178263900066, '2010-01-26 08:00:14', 3, '', true, -1260011116, false, ["YvjL81fR", "4ksR", "fK", "NNqutijP", "", "", "tE"], 110278149, ["", ""]);
CREATE VIEW v0 AS SELECT t1_50.c_1_0, t1_50.c_1_1, t1_50.c_1_2, t1_50.c_1_3, t1_50.c_1_4, t1_50.c_1_5, t1_50.c_1_6, t1_50.c_1_7, t1_50.c_1_8, t1_50.c_1_9, t1_50.c_1_11 FROM t1 AS t1_50 WHERE (NOT (false)) GROUP BY t1_50.c_1_0, t1_50.c_1_1, t1_50.c_1_2, t1_50.c_1_3, t1_50.c_1_4, t1_50.c_1_5, t1_50.c_1_6, t1_50.c_1_7, t1_50.c_1_8, t1_50.c_1_9, t1_50.c_1_11 ORDER BY t1_50.c_1_4, t1_50.c_1_4, t1_50.c_1_11, t1_50.c_1_6, t1_50.c_1_7, t1_50.c_1_3, t1_50.c_1_3, t1_50.c_1_11, t1_50.c_1_8, t1_50.c_1_1;
INSERT INTO t0 VALUES ('2010-01-24 08:00:14', '2010-01-09 08:00:14', '2010-01-26 08:00:14', percentile_hash('-383622571'), 0.479805037760493, 1368484479, false, 723713955, hll_hash(''), 0.5987237936941517, 1.249208943E9, false), ('2010-01-04 08:00:14', '2010-01-14 08:00:14', '2010-01-23 08:00:14', percentile_hash('-532917514'), 0.828062793704816, 759402767, NULL, NULL, hll_hash('0C'), 0.002459879789817987, -1.280109443E9, false);
INSERT INTO t1 VALUES ('2010-01-29 08:00:14', '', 31, 0.5100736301054, '2010-01-03 08:00:14', 3, 'Q', true, -54834903, true, ["y", "v9vb", "gj"], 1055532875, ["6N0", "U5ig", "WRuFzkuCS", "gNzT", "O4", "tGm", "6x", "WS0MQ", "", "S9p", "5", "l", "oP", "g", "u1lOQU6"]);
INSERT INTO t0 VALUES ('2010-01-20 08:00:14', '2010-01-15 08:00:14', '2010-01-16 08:00:14', percentile_hash('172802601'), 0.502486402371113, 359785339, false, 110278149, hll_hash('a3'), NULL, 0.4531095051745787, false), ('2010-01-23 08:00:14', '2010-01-20 08:00:14', '2010-01-25 08:00:14', percentile_hash('166037822'), 0.829193748521890, 1383880049, true, 1503088321, hll_hash('o'), 0.5915825230806343, -6.64929238E8, false);
INSERT INTO t0 VALUES ('2010-01-03 08:00:14', NULL, '2010-01-26 08:00:14', percentile_hash('449189335'), 0.832924277486566, 759402767, true, -664929238, hll_hash('37DU1hXx'), -8.72123264E8, 0.3274199923857031, false), ('2010-01-10 08:00:14', '2010-01-21 08:00:14', '2010-01-14 08:00:14', percentile_hash('435697369'), 0.671843269495538, -623276025, true, 2076343173, hll_hash(''), 7.59402767E8, 0.6397474342226216, true);
INSERT INTO t1 VALUES ('2010-01-27 08:00:14', '', 19, 0.5139492789708, '2010-01-06 08:00:14', 5, 'AK', true, 878223015, true, ["5Nq4g4g", "gfdu", "", "", "Lp3H", "", "", "", "H", "", "xgyK6lwX"], 2064812976, ["lp", "", "PzQBX", "C", "B", "ZTyhy1alh", "aZE"]);
INSERT INTO t0 VALUES ('2009-12-31 08:00:14', NULL, '2010-01-05 08:00:14', percentile_hash('-113847854'), 0.605118036855461, 388532945, false, -615782405, hll_hash(''), 0.9722877260139795, 0.501470743231944, true);
INSERT INTO t1 VALUES ('2010-01-23 08:00:14', '', 25, 0.0651966586455, '2010-01-06 08:00:14', 9, 'R', true, 1982967937, false, ["mEG", "", "v", "", "P", "owG6", "Q", "", "", ""], 2010755685, ["m", "", "9Z7", "9V", "xnsKIM", "", "RasPEIWy", "0", "OzZi", "qqxaN", "koKY3", "ScYhiPaB"]), ('2009-12-31 08:00:14', '', 20, 0.8022973773215, '2010-01-27 08:00:14', 12, 'S', true, -468845387, NULL, ["i", "", "d99", "", "XJK1hk", "h", "p", "Z", "rt", "b", "", "PM7CCw", "b", "2qh", "5p"], -383622571, ["", "oBi", "T", "", "Qbm", "", "", "v", "Yep", "", "mZ", "zx4LX6m", "tdWuS", "", "8G"]), ('2010-01-24 08:00:14', '', 29, 0.3045631604316, '2010-01-02 08:00:14', 12, '', true, 999241393, true, ["NOdqERCS6", "Jqal", "p69", "f0Y", "", "Zy", "", "PTT", "Kx"], -383622571, ["", "", "xq", "T", "Vdl2kITP", "bVW", "k", "", "aP", "p0CL", "Io", "9m", "", "n2f"]);
INSERT INTO t0 VALUES ('2009-12-31 08:00:14', '2010-01-16 08:00:14', '2010-01-12 08:00:14', percentile_hash('-383622571'), 0.012067007330392, -1783457709, false, 1973356944, hll_hash('MFdoe7t'), 0.9990004240220068, 0.5890124913634565, true), ('2010-01-03 08:00:14', '2010-01-21 08:00:14', '2010-01-04 08:00:14', percentile_hash('-123457164'), 0.094034281484676, 1388512516, true, -1713942689, hll_hash('Pps'), 0.3274199923857031, 0.7097373743307518, true), ('2010-01-06 08:00:14', '2010-01-07 08:00:14', '2010-01-08 08:00:14', percentile_hash('293767322'), 0.486582685988883, 55698628, true, -2078861294, hll_hash(''), 0.6053362985683556, 0.29968018109364136, true);
INSERT INTO t0 VALUES ('2010-01-22 08:00:14', '2010-01-02 08:00:14', '2010-01-04 08:00:14', percentile_hash('60088723'), 0.525883653276312, -492235240, true, 2076343173, hll_hash('Zk8I'), 0.20663870702849152, 0.14720372545544358, false), ('2010-01-11 08:00:14', '2010-01-06 08:00:14', '2010-01-07 08:00:14', percentile_hash('-999670000'), 0.751361071634438, 411005371, false, 1085849381, hll_hash(''), 0.47923104041911146, 1.085849381E9, true);
-- SELECT '1969-12-25 01:41:47' FROM  (SELECT v0_56.c_1_0, v0_56.c_1_1, v0_56.c_1_2, v0_56.c_1_3, v0_56.c_1_4, v0_56.c_1_5, v0_56.c_1_6, v0_56.c_1_7, v0_56.c_1_8, v0_56.c_1_9, v0_56.c_1_11 FROM v0 AS v0_56) subv0 LEFT ANTI JOIN t0 AS t0_54 ON subv0.c_1_5 = t0_54.c_0_7 AND ((-1398522340)*(-269827043)) >= ((NULL)/(435697369)) AND subv0.c_1_11 = t0_54.c_0_7 AND subv0.c_1_11 = t0_54.c_0_7;
-- cardinality: 15
-- SELECT '1969-12-25 01:41:47' FROM  (SELECT v0_56.c_1_0, v0_56.c_1_1, v0_56.c_1_2, v0_56.c_1_3, v0_56.c_1_4, v0_56.c_1_5, v0_56.c_1_6, v0_56.c_1_7, v0_56.c_1_8, v0_56.c_1_9, v0_56.c_1_11 FROM v0 AS v0_56) subv0 LEFT ANTI JOIN t0 AS t0_54 ON subv0.c_1_5 = t0_54.c_0_7 AND ((-1398522340)*(-269827043)) >= ((NULL)/(435697369)) AND subv0.c_1_11 = t0_54.c_0_7 AND subv0.c_1_11 = t0_54.c_0_7 WHERE (NOT ((NOT (CAST(((872459142)%(((-420316705)*(-1683869116)))) AS BOOLEAN ) )))) UNION ALL SELECT '1969-12-25 01:41:47' FROM  (SELECT v0_56.c_1_0, v0_56.c_1_1, v0_56.c_1_2, v0_56.c_1_3, v0_56.c_1_4, v0_56.c_1_5, v0_56.c_1_6, v0_56.c_1_7, v0_56.c_1_8, v0_56.c_1_9, v0_56.c_1_11 FROM v0 AS v0_56) subv0 LEFT ANTI JOIN t0 AS t0_54 ON subv0.c_1_5 = t0_54.c_0_7 AND ((-1398522340)*(-269827043)) >= ((NULL)/(435697369)) AND subv0.c_1_11 = t0_54.c_0_7 AND subv0.c_1_11 = t0_54.c_0_7 WHERE (NOT ((NOT ((NOT (CAST(((872459142)%(((-420316705)*(-1683869116)))) AS BOOLEAN ) )))))) UNION ALL SELECT '1969-12-25 01:41:47' FROM  (SELECT v0_56.c_1_0, v0_56.c_1_1, v0_56.c_1_2, v0_56.c_1_3, v0_56.c_1_4, v0_56.c_1_5, v0_56.c_1_6, v0_56.c_1_7, v0_56.c_1_8, v0_56.c_1_9, v0_56.c_1_11 FROM v0 AS v0_56) subv0 LEFT ANTI JOIN t0 AS t0_54 ON subv0.c_1_5 = t0_54.c_0_7 AND ((-1398522340)*(-269827043)) >= ((NULL)/(435697369)) AND subv0.c_1_11 = t0_54.c_0_7 AND subv0.c_1_11 = t0_54.c_0_7 WHERE (((NOT ((NOT (CAST(((872459142)%(((-420316705)*(-1683869116)))) AS BOOLEAN ) ))))) IS NULL);
-- cardinality: 45

--java.lang.AssertionError: the size of the result sets mismatch (0 and 6)!
---- SELECT t0_56.c_0_0 FROM t0 AS t0_56, t1 AS t1_57 GROUP BY t0_56.c_0_2, t0_56.c_0_0;
---- cardinality: 0
---- SELECT t0_56.c_0_0 FROM t0 AS t0_56, t1 AS t1_57 GROUP BY t0_56.c_0_2, t0_56.c_0_0 HAVING MIN( ( t1_57.c_1_2 ) >= ( t1_57.c_1_2 ) )  UNION ALL SELECT t0_56.c_0_0 FROM t0 AS t0_56, t1 AS t1_57 GROUP BY t0_56.c_0_2, t0_56.c_0_0 HAVING (NOT (MIN( ( t1_57.c_1_2 ) >= ( t1_57.c_1_2 ) ) )) UNION ALL SELECT t0_56.c_0_0 FROM t0 AS t0_56, t1 AS t1_57 GROUP BY t0_56.c_0_2, t0_56.c_0_0 HAVING ((MIN( ( t1_57.c_1_2 ) >= ( t1_57.c_1_2 ) ) ) IS NULL);
---- cardinality: 6
--	at sqlancer.ComparatorHelper.assumeResultSetsAreEqual(ComparatorHelper.java:104)
--	at sqlancer.starrocks.oracle.tlp.StarRocksTLPHavingOracle.check(StarRocksTLPHavingOracle.java:56)
--	at sqlancer.ProviderAdapter.generateAndTestDatabase(ProviderAdapter.java:55)
--	at sqlancer.Main$DBMSExecutor.run(Main.java:326)
--	at sqlancer.Main$2.run(Main.java:521)
--	at sqlancer.Main$2.runThread(Main.java:498)
--	at sqlancer.Main$2.run(Main.java:488)
--	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149)
--	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624)
--	at java.lang.Thread.run(Thread.java:748)
---- Time: 2022/10/16 23:07:22
-- Database: h5
-- Database version: 5.1.0
-- seed value: 1665932412897
DROP DATABASE IF EXISTS h5;
CREATE DATABASE h5;
USE h5;
admin set frontend config ("enable_decimal_v3" = "true");
CREATE TABLE t0 (c_0_0 DATE NULL ,c_0_1 DECIMAL(30, 10)  NULL ,c_0_2 DATE NOT NULL ,c_0_3 DECIMAL(15, 5)  NOT NULL ,c_0_4 BOOLEAN NOT NULL ,c_0_5 CHAR(21) REPLACE_IF_NOT_NULL NOT NULL ,c_0_6 BOOLEAN REPLACE NOT NULL ,c_0_7 BOOLEAN REPLACE NULL ,c_0_8 FLOAT MAX NULL ,c_0_9 HLL HLL_UNION NOT NULL ,c_0_10 DECIMAL(28, 2)  MIN NOT NULL , INDEX index1_c_0_2(`c_0_2`) USING BITMAP) AGGREGATE KEY (c_0_0,c_0_1,c_0_2,c_0_3,c_0_4) PARTITION BY RANGE(c_0_2) (START ("2010-01-01") END ("2022-12-31") EVERY (INTERVAL 339 day)) DISTRIBUTED BY HASH (c_0_3) BUCKETS 3 properties("replication_num"="3") ;
CREATE TABLE t1 (c_1_0 BOOLEAN NULL ,c_1_1 SMALLINT SUM NULL ,c_1_2 FLOAT MIN NULL ,c_1_3 SMALLINT REPLACE_IF_NOT_NULL NOT NULL ,c_1_4 DECIMAL(31, 9)  SUM NOT NULL ,c_1_5 PERCENTILE PERCENTILE_UNION NULL ,c_1_6 VARCHAR(1) MIN NULL ) AGGREGATE KEY (c_1_0) DISTRIBUTED BY HASH (c_1_0) BUCKETS 3 properties("replication_num"="3") ;
CREATE TABLE IF NOT EXISTS t1 (c_1_0 VARCHAR(21) NOT NULL,c_1_1 BOOLEAN NOT NULL,c_1_2 DATE NOT NULL,c_1_3 BOOLEAN NOT NULL,c_1_4 DECIMAL(12, 7)  NOT NULL,c_1_5 DATE NOT NULL,c_1_6 BOOLEAN NOT NULL,c_1_7 ARRAY<STRING> NOT NULL) PRIMARY KEY (c_1_0,c_1_1,c_1_2,c_1_3) DISTRIBUTED BY HASH (c_1_2,c_1_0) BUCKETS 3 properties("replication_num"="3") ;
CREATE TABLE t2 (c_2_0 DATETIME NOT NULL,c_2_1 DATE NOT NULL,c_2_2 VARCHAR(11) NOT NULL,c_2_3 VARCHAR(1) NOT NULL,c_2_4 DATE NOT NULL,c_2_5 DATETIME NOT NULL,c_2_6 DATETIME NOT NULL,c_2_7 INT NOT NULL,c_2_8 VARCHAR(1) NOT NULL,c_2_9 BIGINT NOT NULL,c_2_10 ARRAY<STRING> NOT NULL,c_2_11 DOUBLE NOT NULL,c_2_12 ARRAY<STRING> NOT NULL,c_2_13 DECIMAL(12, 11)  NOT NULL,c_2_14 DECIMAL(9, 2)  NOT NULL,c_2_15 ARRAY<STRING> NOT NULL,c_2_16 ARRAY<STRING> NOT NULL,c_2_17 ARRAY<INT> NOT NULL, INDEX index1_c_2_0(`c_2_0`) USING BITMAP) PRIMARY KEY (c_2_0,c_2_1,c_2_2,c_2_3,c_2_4,c_2_5,c_2_6,c_2_7,c_2_8) PARTITION BY RANGE(c_2_1) (START ("2010-01-01") END ("2022-12-31") EVERY (INTERVAL 135 day)) DISTRIBUTED BY HASH (c_2_8,c_2_5,c_2_2,c_2_1,c_2_6,c_2_3,c_2_4,c_2_7,c_2_0) BUCKETS 3 properties("replication_num"="3", "bloom_filter_columns"="c_2_8") ;
TRUNCATE TABLE t0;
DELETE FROM t0 WHERE c_0_2='1969-12-08' AND c_0_4=true;
INSERT INTO t0 VALUES ('2010-01-06', 2516048811.0674870575, '2010-01-11', 22272.37835, false, NULL, false, false, NULL, hll_hash('HczwcITu'), 939270949672947006447998.31), ('2010-01-23', 3716649881.9267906468, '2010-01-17', 69025.75415, true, '', true, false, NULL, hll_hash('rSg'), 298950611755034012517739.94);
INSERT INTO t2 VALUES ('2010-01-14 08:00:14', '2010-01-31', 'h', '', '2010-01-10', '2010-01-01 08:00:14', '2010-01-20 08:00:14', 1, '', 996874355, ["Kv", "iJeDz", "", "g", "dcQAE4Q1"], 9.96874355E8, ["", "", "Q", "tq", "aFyPvSfi", "", "", "", ""], 7E-11, 23253.24, ["vKFUleZXo", "z", "K", ""], [], []);
INSERT INTO t0 VALUES (NULL, 9368968609.0096291485, '2010-01-24', 99747.68287, true, NULL, false, false, 9.96874355E8, hll_hash('PVc'), 323772619848412768206458.19), ('2010-01-30', 2638336845.4241417237, '2010-01-26', 57242.23784, true, NULL, true, NULL, 0.20728488896954445, hll_hash('2b2dLii6'), 159804034799941208078166.78);
INSERT INTO t0 VALUES ('2010-01-12', 3267247884.4305137034, '2010-01-29', 92363.64923, true, '', true, false, 0.04982029037278557, hll_hash('Qd'), 164522478044373920980092.47), (NULL, 2905265066.0352713818, '2010-01-26', 81754.15619, true, '', false, false, 0.04982029037278557, hll_hash('jqi'), 518427844928774694821527.21);
-- SELECT t0_56.c_0_0 FROM t0 AS t0_56, t1 AS t1_57 GROUP BY t0_56.c_0_2, t0_56.c_0_0;
-- cardinality: 0
-- SELECT t0_56.c_0_0 FROM t0 AS t0_56, t1 AS t1_57 GROUP BY t0_56.c_0_2, t0_56.c_0_0 HAVING MIN( ( t1_57.c_1_2 ) >= ( t1_57.c_1_2 ) )  UNION ALL SELECT t0_56.c_0_0 FROM t0 AS t0_56, t1 AS t1_57 GROUP BY t0_56.c_0_2, t0_56.c_0_0 HAVING (NOT (MIN( ( t1_57.c_1_2 ) >= ( t1_57.c_1_2 ) ) )) UNION ALL SELECT t0_56.c_0_0 FROM t0 AS t0_56, t1 AS t1_57 GROUP BY t0_56.c_0_2, t0_56.c_0_0 HAVING ((MIN( ( t1_57.c_1_2 ) >= ( t1_57.c_1_2 ) ) ) IS NULL);
-- cardinality: 6

--- a/fe/fe-core/src/main/java/com/starrocks/sql/optimizer/rule/transformation/PushDownPredicateAggRule.java
+++ b/fe/fe-core/src/main/java/com/starrocks/sql/optimizer/rule/transformation/PushDownPredicateAggRule.java
@@ -39,8 +39,6 @@ public class PushDownPredicateAggRule extends TransformationRule {
             List<ColumnRefOperator> columns = Utils.extractColumnRef(scalar);
 
             // push down predicate
-            // may be some problems. e.g. select count(*) from tbl having false return empty results.
-            // while select count(*) from (select * from tbl where false) t return one row with result of 0.
             if (groupColumns.containsAll(columns) && !columns.isEmpty()) {
                 // remove from filter
                 iter.remove();

--- a/fe/fe-core/src/main/java/com/starrocks/sql/optimizer/rule/transformation/PushDownPredicateAggRule.java
+++ b/fe/fe-core/src/main/java/com/starrocks/sql/optimizer/rule/transformation/PushDownPredicateAggRule.java
@@ -39,10 +39,12 @@ public class PushDownPredicateAggRule extends TransformationRule {
             List<ColumnRefOperator> columns = Utils.extractColumnRef(scalar);
 
             // push down predicate
+            // may be some problems. e.g. select count(*) from tbl having false return empty results.
+            // while select count(*) from (select * from tbl where false) t return one row with result of 0.
             if (groupColumns.containsAll(columns) && !columns.isEmpty()) {
                 // remove from filter
                 iter.remove();
-                // and to predicates
+                // add to push down predicates
                 pushDownPredicates.add(scalar);
             }
         }

--- a/fe/fe-core/src/main/java/com/starrocks/sql/optimizer/rule/transformation/PushDownPredicateScanRule.java
+++ b/fe/fe-core/src/main/java/com/starrocks/sql/optimizer/rule/transformation/PushDownPredicateScanRule.java
@@ -69,8 +69,9 @@ public class PushDownPredicateScanRule extends TransformationRule {
 
         // clone a new scan operator and rewrite predicate.
         Operator.Builder builder = OperatorBuilderFactory.build(logicalScanOperator);
-        LogicalScanOperator newScanOperator = (LogicalScanOperator) builder.withOperator(logicalScanOperator).build();
-        newScanOperator.setPredicate(predicates);
+        LogicalScanOperator newScanOperator = (LogicalScanOperator) builder.withOperator(logicalScanOperator)
+                .setPredicate(predicates)
+                .build();
         newScanOperator.buildColumnFilters(predicates);
         Map<ColumnRefOperator, ScalarOperator> projectMap =
                 newScanOperator.getOutputColumns().stream()

--- a/fe/fe-core/src/main/java/com/starrocks/sql/optimizer/rule/transformation/RemoveAggregationFromAggTable.java
+++ b/fe/fe-core/src/main/java/com/starrocks/sql/optimizer/rule/transformation/RemoveAggregationFromAggTable.java
@@ -146,6 +146,8 @@ public class RemoveAggregationFromAggTable extends TransformationRule {
                 ScalarOperator rewrittenOperator = rewriter.rewrite(entry.getValue());
                 newProjectMap.put(entry.getKey(), rewrittenOperator);
             }
+        } else {
+            newProjectMap = projectMap;
         }
 
         LogicalProjectOperator projectOperator = new LogicalProjectOperator(newProjectMap);

--- a/fe/fe-core/src/main/java/com/starrocks/sql/optimizer/rule/transformation/RemoveAggregationFromAggTable.java
+++ b/fe/fe-core/src/main/java/com/starrocks/sql/optimizer/rule/transformation/RemoveAggregationFromAggTable.java
@@ -15,6 +15,7 @@ import com.starrocks.sql.optimizer.OptExpression;
 import com.starrocks.sql.optimizer.OptimizerContext;
 import com.starrocks.sql.optimizer.operator.OperatorType;
 import com.starrocks.sql.optimizer.operator.logical.LogicalAggregationOperator;
+import com.starrocks.sql.optimizer.operator.logical.LogicalFilterOperator;
 import com.starrocks.sql.optimizer.operator.logical.LogicalOlapScanOperator;
 import com.starrocks.sql.optimizer.operator.logical.LogicalProjectOperator;
 import com.starrocks.sql.optimizer.operator.pattern.Pattern;
@@ -51,9 +52,6 @@ public class RemoveAggregationFromAggTable extends TransformationRule {
     @Override
     public boolean check(OptExpression input, OptimizerContext context) {
         OptExpression scanExpression = input.getInputs().get(0);
-        if (!(scanExpression.getOp() instanceof LogicalOlapScanOperator)) {
-            return false;
-        }
         LogicalOlapScanOperator scanOperator = (LogicalOlapScanOperator) scanExpression.getOp();
         if (scanOperator.getProjection() != null) {
             return false;
@@ -129,19 +127,28 @@ public class RemoveAggregationFromAggTable extends TransformationRule {
             // in this case, CallOperator must have only one child ColumnRefOperator
             projectMap.put(entry.getKey(), entry.getValue().getChild(0));
         }
+
+        Map<ColumnRefOperator, ScalarOperator> newProjectMap = Maps.newHashMap();
+        ReplaceColumnRefRewriter rewriter = new ReplaceColumnRefRewriter(projectMap);
+
+        OptExpression newChildOpt = input.inputAt(0);
+        if (aggregationOperator.getPredicate() != null) {
+            // rewrite the having predicate. replace the aggFunc by the columnRef
+            ScalarOperator newPredicate = rewriter.rewrite(aggregationOperator.getPredicate());
+            LogicalFilterOperator filterOperator = new LogicalFilterOperator(newPredicate);
+            newChildOpt = OptExpression.create(filterOperator, newChildOpt);
+        }
+
         if (aggregationOperator.getProjection() != null) {
-            Map<ColumnRefOperator, ScalarOperator> newProjectMap =
-                    Maps.newHashMap(aggregationOperator.getProjection().getColumnRefMap());
-            ReplaceColumnRefRewriter rewriter = new ReplaceColumnRefRewriter(projectMap);
             for (Map.Entry<ColumnRefOperator, ScalarOperator> entry :
                     aggregationOperator.getProjection().getColumnRefMap().entrySet()) {
+                // rewrite the projection of this agg. replace the aggFunc by the columnRef
                 ScalarOperator rewrittenOperator = rewriter.rewrite(entry.getValue());
                 newProjectMap.put(entry.getKey(), rewrittenOperator);
             }
-            projectMap.clear();
-            projectMap.putAll(newProjectMap);
         }
-        LogicalProjectOperator projectOperator = new LogicalProjectOperator(projectMap);
-        return Lists.newArrayList(OptExpression.create(projectOperator, input.getInputs().get(0)));
+
+        LogicalProjectOperator projectOperator = new LogicalProjectOperator(newProjectMap);
+        return Lists.newArrayList(OptExpression.create(projectOperator, newChildOpt));
     }
 }

--- a/fe/fe-core/src/main/java/com/starrocks/sql/optimizer/rule/transformation/RemoveAggregationFromAggTable.java
+++ b/fe/fe-core/src/main/java/com/starrocks/sql/optimizer/rule/transformation/RemoveAggregationFromAggTable.java
@@ -137,10 +137,11 @@ public class RemoveAggregationFromAggTable extends TransformationRule {
             ScalarOperator newPredicate = rewriter.rewrite(aggregationOperator.getPredicate());
             LogicalOlapScanOperator scanOperator = (LogicalOlapScanOperator) newChildOpt.getOp();
             LogicalOlapScanOperator.Builder builder = new LogicalOlapScanOperator.Builder();
-            List<ScalarOperator> pushDownPredicates = Lists.newArrayList(newPredicate);
+            List<ScalarOperator> pushDownPredicates = Lists.newArrayList();
             if (scanOperator.getPredicate() != null) {
                 pushDownPredicates.add(scanOperator.getPredicate());
             }
+            pushDownPredicates.add(newPredicate);
             scanOperator = builder.withOperator(scanOperator)
                     .setPredicate(Utils.compoundAnd(pushDownPredicates))
                     .build();

--- a/fe/fe-core/src/main/java/com/starrocks/sql/optimizer/rule/transformation/RemoveAggregationFromAggTable.java
+++ b/fe/fe-core/src/main/java/com/starrocks/sql/optimizer/rule/transformation/RemoveAggregationFromAggTable.java
@@ -13,9 +13,9 @@ import com.starrocks.catalog.PartitionInfo;
 import com.starrocks.catalog.RangePartitionInfo;
 import com.starrocks.sql.optimizer.OptExpression;
 import com.starrocks.sql.optimizer.OptimizerContext;
+import com.starrocks.sql.optimizer.Utils;
 import com.starrocks.sql.optimizer.operator.OperatorType;
 import com.starrocks.sql.optimizer.operator.logical.LogicalAggregationOperator;
-import com.starrocks.sql.optimizer.operator.logical.LogicalFilterOperator;
 import com.starrocks.sql.optimizer.operator.logical.LogicalOlapScanOperator;
 import com.starrocks.sql.optimizer.operator.logical.LogicalProjectOperator;
 import com.starrocks.sql.optimizer.operator.pattern.Pattern;
@@ -135,8 +135,16 @@ public class RemoveAggregationFromAggTable extends TransformationRule {
         if (aggregationOperator.getPredicate() != null) {
             // rewrite the having predicate. replace the aggFunc by the columnRef
             ScalarOperator newPredicate = rewriter.rewrite(aggregationOperator.getPredicate());
-            LogicalFilterOperator filterOperator = new LogicalFilterOperator(newPredicate);
-            newChildOpt = OptExpression.create(filterOperator, newChildOpt);
+            LogicalOlapScanOperator scanOperator = (LogicalOlapScanOperator) newChildOpt.getOp();
+            LogicalOlapScanOperator.Builder builder = new LogicalOlapScanOperator.Builder();
+            List<ScalarOperator> pushDownPredicates = Lists.newArrayList(newPredicate);
+            if (scanOperator.getPredicate() != null) {
+                pushDownPredicates.add(scanOperator.getPredicate());
+            }
+            scanOperator = builder.withOperator(scanOperator)
+                    .setPredicate(Utils.compoundAnd(pushDownPredicates))
+                    .build();
+            newChildOpt = OptExpression.create(scanOperator, newChildOpt.getInputs());
         }
 
         if (aggregationOperator.getProjection() != null) {

--- a/fe/fe-core/src/test/java/com/starrocks/sql/plan/RemoveAggTest.java
+++ b/fe/fe-core/src/test/java/com/starrocks/sql/plan/RemoveAggTest.java
@@ -48,6 +48,7 @@ public class RemoveAggTest extends PlanTestBase {
         sqlList.add("select max(v1) + min(v2) from test_agg group by k1, k2, k3 having (max(v1) + min(v2)) is not null");
         sqlList.add("select max(v1) + min(v2) from test_agg group by k1, k2, k3 " +
                 "having ((max(v1) + min(v2)) is not null or k1 > 4)");
+        sqlList.add("select k1, k2 from test_agg group by k1, k2, k3 having k1 > 1");
         return sqlList.stream().map(e -> Arguments.of(e));
     }
 

--- a/fe/fe-core/src/test/java/com/starrocks/sql/plan/RemoveAggTest.java
+++ b/fe/fe-core/src/test/java/com/starrocks/sql/plan/RemoveAggTest.java
@@ -25,6 +25,7 @@ public class RemoveAggTest extends PlanTestBase {
     @MethodSource("removeAggSqlCases")
     void removeAggTest(String sql) throws Exception {
         String plan = getFragmentPlan(sql);
+        System.out.println(plan);
         assertNotContains(plan, "AGGREGATE");
         assertContains(plan, "PREDICATES");
         assertContains(plan, "PREAGGREGATION: OFF");

--- a/fe/fe-core/src/test/java/com/starrocks/sql/plan/RemoveAggTest.java
+++ b/fe/fe-core/src/test/java/com/starrocks/sql/plan/RemoveAggTest.java
@@ -11,6 +11,7 @@ import org.junit.jupiter.params.provider.Arguments;
 import org.junit.jupiter.params.provider.MethodSource;
 
 import java.util.List;
+import java.util.Locale;
 import java.util.stream.Stream;
 
 public class RemoveAggTest extends PlanTestBase {
@@ -26,6 +27,7 @@ public class RemoveAggTest extends PlanTestBase {
     void removeAggTest(String sql) throws Exception {
         String plan = getFragmentPlan(sql);
         assertNotContains(plan, "AGGREGATE");
+        assertContains(plan.toLowerCase(Locale.ROOT), "predicates");
         assertContains(plan, "PREAGGREGATION: OFF");
     }
 
@@ -40,7 +42,6 @@ public class RemoveAggTest extends PlanTestBase {
 
     private static Stream<Arguments> removeAggSqlCases() {
         List<String> sqlList = Lists.newArrayList();
-        sqlList.add("select max(v1) from test_agg group by k1, k2, k3;");
         sqlList.add("select max(v1) from test_agg group by k1, k2, k3 having(max(v1) = 1)");
         sqlList.add("select max(v1), min(v2) from test_agg group by k1, k2, k3 " +
                 "having (max(v1) > min(v2) or abs(max(v1)) is null)");
@@ -49,6 +50,7 @@ public class RemoveAggTest extends PlanTestBase {
         sqlList.add("select max(v1) + min(v2) from test_agg group by k1, k2, k3 " +
                 "having ((max(v1) + min(v2)) is not null or k1 > 4)");
         sqlList.add("select k1, k2 from test_agg group by k1, k2, k3 having k1 > 1");
+        sqlList.add("select k1, k2 from test_agg group by k1, k2, k3 having k1 > 1 and k2 < 1 and k1 + k2 >3");
         return sqlList.stream().map(e -> Arguments.of(e));
     }
 

--- a/fe/fe-core/src/test/java/com/starrocks/sql/plan/RemoveAggTest.java
+++ b/fe/fe-core/src/test/java/com/starrocks/sql/plan/RemoveAggTest.java
@@ -11,7 +11,6 @@ import org.junit.jupiter.params.provider.Arguments;
 import org.junit.jupiter.params.provider.MethodSource;
 
 import java.util.List;
-import java.util.Locale;
 import java.util.stream.Stream;
 
 public class RemoveAggTest extends PlanTestBase {
@@ -27,7 +26,7 @@ public class RemoveAggTest extends PlanTestBase {
     void removeAggTest(String sql) throws Exception {
         String plan = getFragmentPlan(sql);
         assertNotContains(plan, "AGGREGATE");
-        assertContains(plan.toLowerCase(Locale.ROOT), "predicates");
+        assertContains(plan, "PREDICATES");
         assertContains(plan, "PREAGGREGATION: OFF");
     }
 
@@ -42,15 +41,17 @@ public class RemoveAggTest extends PlanTestBase {
 
     private static Stream<Arguments> removeAggSqlCases() {
         List<String> sqlList = Lists.newArrayList();
-        sqlList.add("select max(v1) from test_agg group by k1, k2, k3 having(max(v1) = 1)");
+        sqlList.add("select max(v1) from test_agg group by k1, k2, k3 having (max(v1) = 1)");
         sqlList.add("select max(v1), min(v2) from test_agg group by k1, k2, k3 " +
                 "having (max(v1) > min(v2) or abs(max(v1)) is null)");
-        sqlList.add("select max(v1), min(v2), abs(123) from test_agg group by k1, k2, k3 having (max(v1) + min(v2) > sum(v3))");
+        sqlList.add("select max(v1), min(v2), abs(123) from test_agg group by k1, k2, k3" +
+                " having (max(v1) + min(v2) > sum(v3))");
         sqlList.add("select max(v1) + min(v2) from test_agg group by k1, k2, k3 having (max(v1) + min(v2)) is not null");
         sqlList.add("select max(v1) + min(v2) from test_agg group by k1, k2, k3 " +
                 "having ((max(v1) + min(v2)) is not null or k1 > 4)");
         sqlList.add("select k1, k2 from test_agg group by k1, k2, k3 having k1 > 1");
-        sqlList.add("select k1, k2 from test_agg group by k1, k2, k3 having k1 > 1 and k2 < 1 and k1 + k2 >3");
+        sqlList.add("select k1, k2 from test_agg group by k1, k2, k3 having k1 > 1 and k2 < 1 " +
+                "and k1 + k2 >3 and max(v1) = 1");
         return sqlList.stream().map(e -> Arguments.of(e));
     }
 
@@ -61,6 +62,9 @@ public class RemoveAggTest extends PlanTestBase {
         sqlList.add("select max(v1) from test_agg group by k1");
         sqlList.add("select max(v1) from test_agg group by abs(k1), k1, k2, k3");
         sqlList.add("select hll_union_agg(h1) from test_agg group by k1, k2, k3");
+        sqlList.add("select max(v1) from t0 group by v2 having max(v1) > 2");
+        sqlList.add("select max(v1) from test_agg group by k1, k2, k3 having (min(v1) = 1)");
+        sqlList.add("select k1, k2 from test_agg group by k1, k2, k3 having max(v1) > 1 or min(v1) < 1");
         return sqlList.stream().map(e -> Arguments.of(e));
     }
 

--- a/fe/fe-core/src/test/java/com/starrocks/sql/plan/RemoveAggTest.java
+++ b/fe/fe-core/src/test/java/com/starrocks/sql/plan/RemoveAggTest.java
@@ -1,0 +1,66 @@
+// This file is licensed under the Elastic License 2.0. Copyright 2021-present, StarRocks Inc.
+
+package com.starrocks.sql.plan;
+
+
+import com.google.common.collect.Lists;
+import com.starrocks.common.FeConstants;
+import org.junit.jupiter.api.BeforeAll;
+import org.junit.jupiter.params.ParameterizedTest;
+import org.junit.jupiter.params.provider.Arguments;
+import org.junit.jupiter.params.provider.MethodSource;
+
+import java.util.List;
+import java.util.stream.Stream;
+
+public class RemoveAggTest extends PlanTestBase{
+
+    @BeforeAll
+    public static void beforeClass() throws Exception {
+        PlanTestBase.beforeClass();
+        FeConstants.runningUnitTest = true;
+    }
+
+    @ParameterizedTest(name = "sql_{index}: {0}.")
+    @MethodSource("removeAggSqlCases")
+    void removeAggTest(String sql) throws Exception {
+        String plan = getFragmentPlan(sql);
+        assertNotContains(plan, "AGGREGATE");
+        assertContains(plan, "PREAGGREGATION: OFF");
+    }
+
+
+    @ParameterizedTest(name = "sql_{index}: {0}.")
+    @MethodSource("remainAggSqlCases")
+    void remainAggTest(String sql) throws Exception {
+        String plan = getFragmentPlan(sql);
+        assertContains(plan, "AGGREGATE");
+    }
+
+
+    private static Stream<Arguments> removeAggSqlCases() {
+        List<String> sqlList = Lists.newArrayList();
+        sqlList.add("select max(v1) from test_agg group by k1, k2, k3;");
+        sqlList.add("select max(v1) from test_agg group by k1, k2, k3 having(max(v1) = 1)");
+        sqlList.add("select max(v1), min(v2) from test_agg group by k1, k2, k3 " +
+                "having (max(v1) > min(v2) or abs(max(v1)) is null)");
+        sqlList.add("select max(v1), min(v2), abs(123) from test_agg group by k1, k2, k3 having (max(v1) + min(v2) > sum(v3))");
+        sqlList.add("select max(v1) + min(v2) from test_agg group by k1, k2, k3 having (max(v1) + min(v2)) is not null");
+        sqlList.add("select max(v1) + min(v2) from test_agg group by k1, k2, k3 " +
+                "having ((max(v1) + min(v2)) is not null or k1 > 4)");
+        return sqlList.stream().map(e -> Arguments.of(e));
+    }
+
+    private static Stream<Arguments> remainAggSqlCases() {
+        List<String> sqlList = Lists.newArrayList();
+        sqlList.add("select max(v1) from t0 group by v2");
+        sqlList.add("select min(v1) from test_agg group by k1, k2, k3");
+        sqlList.add("select max(v1) from test_agg group by k1");
+        sqlList.add("select max(v1) from test_agg group by abs(k1), k1, k2, k3");
+        sqlList.add("select hll_union_agg(h1) from test_agg group by k1, k2, k3");
+        return sqlList.stream().map(e -> Arguments.of(e));
+    }
+
+
+
+}

--- a/fe/fe-core/src/test/java/com/starrocks/sql/plan/RemoveAggTest.java
+++ b/fe/fe-core/src/test/java/com/starrocks/sql/plan/RemoveAggTest.java
@@ -2,9 +2,9 @@
 
 package com.starrocks.sql.plan;
 
-
 import com.google.common.collect.Lists;
 import com.starrocks.common.FeConstants;
+import org.junit.jupiter.api.AfterAll;
 import org.junit.jupiter.api.BeforeAll;
 import org.junit.jupiter.params.ParameterizedTest;
 import org.junit.jupiter.params.provider.Arguments;
@@ -13,7 +13,7 @@ import org.junit.jupiter.params.provider.MethodSource;
 import java.util.List;
 import java.util.stream.Stream;
 
-public class RemoveAggTest extends PlanTestBase{
+public class RemoveAggTest extends PlanTestBase {
 
     @BeforeAll
     public static void beforeClass() throws Exception {
@@ -61,6 +61,8 @@ public class RemoveAggTest extends PlanTestBase{
         return sqlList.stream().map(e -> Arguments.of(e));
     }
 
-
-
+    @AfterAll
+    public static void afterClass() {
+        FeConstants.runningUnitTest = false;
+    }
 }

